*** mswin31.c	Sat Feb 14 19:24:22 2004
--- mswin31.c	Mon May 27 22:15:54 2002
***************
*** 514,520 ****
      UseHourGlass = True;
      DisableMenu = False;
  
!     for( i=0; i<8; i++ )
           DialValue[i] = 0.0;
  
      ULut[0] = True;
--- 510,516 ----
      UseHourGlass = True;
      DisableMenu = False;
  
!     for( i=0; i<10; i++ )
           DialValue[i] = 0.0;
  
      ULut[0] = True;
***************
*** 522,527 ****
--- 518,524 ----
      XRange = DefaultWide;   WRange = XRange>>1;
      YRange = DefaultHigh;   HRange = YRange>>1;
      Range = MinFun(XRange,YRange);
+     ZRange = 20000;
      
      rect.top  = 0;   rect.bottom = YRange;
      rect.left = 0;   rect.right  = XRange;
