 <!-- manual page source format generated by PolyglotMan v3.0.8+XFree86, -->
<!-- available via anonymous ftp from ftp.cs.berkeley.edu:/ucb/people/phelps/tcltk/rman.tar.Z -->

<HTML>
<HEAD>
<TITLE>RASMOL(1) manual page</TITLE>
</HEAD>
<BODY BGCOLOR="#efefef" TEXT="black" LINK="blue" VLINK="#551A8B" ALINK="red">
<A HREF="#toc">Table of Contents</A><P>

<H2><A NAME="sect0" HREF="#toc0">Name</A></H2>
rasmol - Molecular Graphics Visualisation Tool v2.7.2.1 <P>
 
<H2><A NAME="sect1" HREF="#toc1">Synopsis</A></H2>
<B>rasmol</B>
[<B>-nodiplay</B>] [[<B>-</B><I>format</I>] <I>filename</I>] [<B>-script </B><I>scriptfile</I>] <P>
 
<H2><A NAME="sect2" HREF="#toc2">Formats</A></H2>
 
<DL>

<DT><B>-pdb</B> </DT>
<DD>Protein
Data Bank </DD>

<DT><B>-mdl</B> </DT>
<DD>MDL's MOL File Format </DD>

<DT><B>-mol2</B> </DT>
<DD>Tripos' Sybyl MOL2 Format </DD>

<DT><B>-xyz</B> </DT>
<DD>MSC's
XYZ (XMol) Format </DD>

<DT><B>-mopac</B> </DT>
<DD>MOPAC Input or Output File Format </DD>

<DT><B>-alchemy</B> </DT>
<DD>Alchemy
File Format </DD>

<DT><B>-charmm</B> </DT>
<DD>CHARMm File Format </DD>

<DT><B>-cif</B> </DT>
<DD>IUCr CIF or CIF File Format 
</DD>
</DL>

<H2><A NAME="sect3" HREF="#toc3">Notices</A></H2>
<P>
 This software has been created from several sources.  Much of the
code is from RasMol 2.6, as created by Roger Sayle. The torsion angle code,
new POVRAY3 code and other features are derived from the RasMol2.6x1 revisions
by Arne Mueller. The Ramachandran printer plot code was derived from fisipl
created by Frances C. Bernstein.  See the Protein Data Bank program tape.
<P>
 The code to display multiple molecules and to allow bond rotation is derived
in large part from the UCB mods by Gary Grossman and Marco Molinaro, included
with permission of Eileen Lewis of the ModularCHEM Consortium. <P>
 The CIF
modifications make use of a library based in part on CBFlib by Paul J. Ellis
and Herbert J. Bernstein. Parts of CBFlib is loosely based on the CIFPARSE
software package from the NDB at Rutgers university. Please type the RasMol
commands <B>help copying,</B> <B>help general,</B> <B>help IUCR,</B> <B>help CBFlib,</B>  and<BR>
 <B>help CIFPARSE</B> for applicable notices.  Please type <B>help copyright</B> for copyright
notices.  If you use RasMol V2.6 or an earlier version, type the RasMol command
<B>help oldnotice.</B> <P>
 
<DL>

<DT><B>Copying RasMol</B> </DT>
<DD>This version is based in large part on RasMol
version 2.7.2, RasMol version 2.7.1.1 and RasTop version 1.3 and indirectly on
the RasMol 2.5-ucb and 2.6-ucb versions and version 2.6_CIF.2, RasMol 2.6x1 and
RasMol_2.6.4. <P>
 If you are not going to make changes to RasMol, you are not
only permitted to freely make copies and distribute them, you are encouraged
to do so, provided you do the following: <P>
   1.  Either include the complete
documentation, especially the file<BR>
 NOTICE, with what you distribute or provide a clear indication where people
can get a copy of the documentation; and <P>
   2.  Please give credit where
credit is due citing the version and<BR>
 original authors properly; and <P>
   3.  Please do not give anyone the impression
that the original<BR>
 authors are providing a warranty of any kind. <P>
 If you would like to use
major pieces of RasMol in some other program, make modifications to RasMol,
or in some other way make what a lawyer would call a "derived work", you
are not only permitted to do so, you are encouraged to do so. In addition
to the things we discussed above, please do the following: <P>
   4.  Please
explain in your documentation how what you did differs<BR>
 from this version of RasMol; and <P>
   5.  Please make your modified source
code available.<BR>
 <P>
 This version of RasMol is _not_ in the public domain, but it is given
freely to the community in the hopes of advancing science.  If you make
changes, please make them in a responsible manner, and please offer us
the opportunity to include those changes in future versions of RasMol. <P>

</DD>

<DT><B>General Notice</B> </DT>
<DD>The following notice applies to this work as a whole and
to the works included within it: <P>
 * Creative endeavors depend on the lively
exchange of ideas. There are laws and customs which establish rights and
responsibilities for authors and the users of what authors create. This
notice is not intended to prevent you from using the software and documents
in this package, but to ensure that there are no misunderstandings about
terms and conditions of such use. <P>
 * Please read the following notice carefully.
If you do not understand any portion of this notice, please seek appropriate
professional legal advice before making use of the software and documents
included in this software package. In addition to whatever other steps you
may be obliged to take to respect the intellectual property rights of the
various parties involved, if you do make use of the software and documents
in this package, please give credit where credit is due by citing this
package, its authors and the URL or other source from which you obtained
it, or equivalent primary references in the literature with the same authors.
<P>
 * Some of the software and documents included within this software package
are the intellectual property of various parties, and placement in this
package does not in any way imply that any such rights have in any way
been waived or diminished. <P>
 * With respect to any software or documents
for which a copyright exists, ALL RIGHTS ARE RESERVED TO THE OWNERS OF
SUCH COPYRIGHT. <P>
 * Even though the authors of the various documents and
software found here have made a good faith effort to ensure that the documents
are correct and that the software performs according to its documentation,
and we would greatly appreciate hearing of any problems you may encounter,
the programs and documents and any files created by the programs are provided
**AS IS** without any warranty as to correctness, merchantability or fitness
for any particular or general use. <P>
 * THE RESPONSIBILITY FOR ANY ADVERSE
CONSEQUENCES FROM THE USE OF PROGRAMS OR DOCUMENTS OR ANY FILE OR FILES
CREATED BY USE OF THE PROGRAMS OR DOCUMENTS LIES SOLELY WITH THE USERS
OF THE PROGRAMS OR DOCUMENTS OR FILE OR FILES AND NOT WITH AUTHORS OF THE
PROGRAMS OR DOCUMENTS. <P>
 Subject to your acceptance of the conditions stated
above, and your respect for the terms and conditions stated in the notices
below, if you are not going to make any modifications or create derived
works, you are given permission to freely copy and distribute this package,
provided you do the following: <P>
 1.  Either include the complete documentation,
especially the file NOTICE, with what you distribute or provide a clear
indication where people can get a copy of the documentation; and <P>
 2.  Give
credit where credit is due citing the version and original authors properly;
<P>
and <P>
 3.  Do not give anyone the impression that the original authors are
providing a warranty of any kind. <P>
 In addition, you may also modify this
package and create derived works provided you do the following: <P>
 4.  Explain
in your documentation how what you did differs from this version of RasMol;
<P>
and <P>
 5.  Make your modified source code available. <P>
 </DD>

<DT><B>RasMol V2.6 Notice</B> </DT>
<DD>The
following notice applies to RasMol V 2.6 and older RasMol versions. <P>
 Information
in this document is subject to change without notice and does not represent
a commitment on the part of the supplier. This package is sold/distributed
subject to the condition that it shall not, by way of trade or otherwise,
be lent, re-sold, hired out or otherwise circulated without the supplier's
prior consent, in any form of packaging or cover other than that in which
it was produced. No part of this manual or accompanying software may be
reproduced, stored in a retrieval system on optical or magnetic disk, tape
or any other medium, or transmitted in any form or by any means, electronic,
mechanical, photocopying, recording or otherwise for any purpose other
than the purchaser's personal use. <P>
 This product is not to be used in the
planning, construction, maintenance, operation or use of any nuclear facility
nor the flight, navigation or communication of aircraft or ground support
equipment. The author shall not be liable, in whole or in part, for any
claims or damages arising from such use, including death, bankruptcy or
outbreak of war. <P>
 </DD>

<DT><B>IUCR Policy</B> </DT>
<DD><B>The IUCr Policy for the Protection and the
Promotion of the STAR File and</B> <B>CIF Standards for Exchanging and Archiving
Electronic Data.</B> <P>
 <B>Overview</B> <P>
 The Crystallographic Information File (CIF)[1]
is a standard for information interchange promulgated by the International
Union of Crystallography (IUCr). CIF (Hall, Allen &amp; Brown, 1991) is the recommended
method for submitting publications to Acta Crystallographica Section C
and reports of crystal structure determinations to other sections of Acta
Crystallographica and many other journals. The syntax of a CIF is a subset
of the more general STAR File[2] format. The CIF and STAR File approaches
are used increasingly in the structural sciences for data exchange and
archiving, and are having a significant influence on these activities in
other fields. <P>
 <B>Statement of intent</B> <P>
 The IUCr's interest in the STAR File
is as a general data interchange standard for science, and its interest
in the CIF, a conformant derivative of the STAR File, is as a concise data
exchange and archival standard for crystallography and structural science.
<P>
 <B>Protection of the standards</B> <P>
 To protect the STAR File and the CIF as standards
for interchanging and archiving electronic data, the IUCr, on behalf of
the scientific community, <P>
    * holds the copyrights on the standards themselves,<BR>
 <P>
    * owns the associated trademarks and service marks, and<BR>
 <P>
    * holds a patent on the STAR File.<BR>
 <P>
 These intellectual property rights relate solely to the interchange formats,
not to the data contained therein, nor to the software used in the generation,
access or manipulation of the data. <P>
 <B>Promotion of the standards</B> <P>
 The sole
requirement that the IUCr, in its protective role, imposes on software
purporting to process STAR File or CIF data is that the following conditions
be met prior to sale or distribution. <P>
    * Software claiming to read files
written to either the STAR File or the<BR>
 CIF standard must be able to extract the pertinent data from a file conformant
to the STAR File syntax, or the CIF syntax, respectively. <P>
    * Software
claiming to write files in either the STAR File, or the CIF,<BR>
 standard must produce files that are conformant to the STAR File syntax,
or the CIF syntax, respectively. <P>
    * Software claiming to read definitions
from a specific data dictionary<BR>
 approved by the IUCr must be able to extract any pertinent definition
which is conformant to the dictionary definition language (DDL)[3] associated
with that dictionary. <P>
 The IUCr, through its Committee on CIF Standards,
will assist any developer to verify that software meets these conformance
conditions. <P>
 <B>Glossary of terms</B> <P>
 [1] CIF: <P>
 is a data file conformant to the
<P>
file syntax defined at <A HREF="http://www.iucr.org/iucr-top/cif/spec/index.html">http://www.iucr.org/iucr-top/cif/spec/index.html</A>
 <P>
 [2]
STAR File: <P>
 is a data file conformant to the file syntax defined at <A HREF="http://www.iucr.org/iucr-top/cif/spec/star/index.html">http://www.iucr.org/iucr-top/cif/spec/star/index.html</A>

<P>
 [3] DDL: <P>
 is a language used in a data dictionary to define data items
in terms of "attributes". Dictionaries currently approved by the IUCr, and
the DDL versions used to construct these dictionaries, are listed at <A HREF="http://www.iucr.org/iucr-top/cif/spec/ddl/index.html">http://www.iucr.org/iucr-top/cif/spec/ddl/index.html</A>

<P>
 Last modified: 30 September 2000 <P>
 IUCr Policy Copyright (C) 2000 International
<P>
Union of Crystallography <P>
 </DD>

<DT><B>CBFLIB</B> </DT>
<DD>The following Disclaimer Notice applies
to CBFlib V0.1, from which this code in part is derived. <P>
 * The items furnished
herewith were developed under the sponsorship of the U.S. Government. Neither
the U.S., nor the U.S. D.O.E., nor the Leland Stanford Junior University, nor
their employees, makes any warranty, express or implied, or assumes any
liability or responsibility for accuracy, completeness or usefulness of
any information, apparatus, product or process disclosed, or represents
that its use will not infringe privately-owned rights. Mention of any product,
its manufacturer, or suppliers shall not, nor is it intended to, imply
approval, disapproval, or fitness for any particular use. The U.S. and the
University at all times retain the right to use and disseminate the furnished
items for any purpose whatsoever. <P>
 Notice 91 02 01 <P>
 </DD>

<DT><B>CIFPARSE</B> </DT>
<DD>Portions of
this software are loosely based on the CIFPARSE software package from the
NDB at Rutgers University.  See <P>
    <A HREF="http://ndbserver.rutgers.edu/NDB/mmcif/software">http://ndbserver.rutgers.edu/NDB/mmcif/software</A>
<BR>
 <P>
 CIFPARSE is part of the NDBQUERY application, a program component of
the Nucleic Acid Database Project [ H. M. Berman, W. K. Olson, D. L. Beveridge,
J. K. Westbrook, A. Gelbin, T. Demeny, S. H. Shieh, A. R. Srinivasan, and B. Schneider.
(1992). The Nucleic Acid Database: A Comprehensive Relational Database of
Three-Dimensional Structures of Nucleic Acids. Biophys J., 63, 751-759.], whose
cooperation is gratefully acknowledged, especially in the form of design
concepts created by J. Westbrook. <P>
 Please be aware of the following notice
in the CIFPARSE API: <P>
 This software is provided WITHOUT WARRANTY OF MERCHANTABILITY
OR FITNESS FOR A PARTICULAR PURPOSE OR ANY OTHER WARRANTY, EXPRESS OR IMPLIED.
RUTGERS MAKE NO REPRESENTATION OR WARRANTY THAT THE SOFTWARE WILL NOT INFRINGE
ANY PATENT, COPYRIGHT OR OTHER PROPRIETARY RIGHT. <P>
 </DD>
</DL>

<H2><A NAME="sect4" HREF="#toc4">Description</A></H2>
RasMol is
a molecular graphics program intended for the visualisation of proteins,
nucleic acids and small molecules. The program is aimed at display, teaching
and generation of publication quality images.   RasMol runs on wide range
of architectures and operating systems including Microsoft Windows, Apple
Macintosh, UNIX and VMS systems. UNIX and VMS versions require an 8, 24
or 32 bit colour X Windows display (X11R4 or later).  The X Windows version
of RasMol provides optional support for a hardware dials box and accelerated
shared memory communication (via the XInput and MIT-SHM extensions) if available
on the current X Server. <P>
 The program reads in a molecule coordinate file
and interactively displays the molecule on the screen in a variety of colour
schemes and molecule representations. Currently available representations
include depth-cued wireframes, 'Dreiding' sticks, spacefilling (CPK) spheres,
ball and stick, solid and strand biomolecular ribbons, atom labels and
dot surfaces. <P>
 Up to 5 molecules may be loaded and displayed at once.  Any
one or all of  the molecules may be rotated and translated.<BR>
 <P>
 The RasMol help facility can be accessed by typing "help &lt;topic&gt;" or "help
&lt;topic&gt; &lt;subtopic&gt;" from the command line. A complete list of RasMol commands
may be displayed by typing "help commands". A single question mark may also
be used to abbreviate the keyword "help".  Please type "help notices" for
important notices. <P>
 
<H2><A NAME="sect5" HREF="#toc5">Commands</A></H2>
RasMol allows the execution of interactive commands
typed at the <B>RasMol&gt;</B> prompt in the terminal window. Each command must be
given on a separate line. Keywords are case insensitive and may be entered
in either upper or lower case letters. All whitespace characters are ignored
except to separate keywords and their arguments. <P>
 The commands/keywords
currently recognised by RasMol are given below. <P>
 
<DL>

<DT><B>Backbone</B> </DT>
<DD>The RasMol <B>backbone</B>
command permits the representation of a polypeptide backbone as a series
of bonds connecting the adjacent alpha carbons of each amino acid in a
chain. The display of these backbone 'bonds' is turned on and off by the command
parameter in the same way as with the <B>wireframe</B> command. The command <B>backbone
off</B> turns off the selected 'bonds', and <B>backbone on</B> or with a number turns
them on. The number can be used to specify the cylinder radius of the representation
in either Angstrom or RasMol units. A parameter value of 500 (2.0 Angstroms)
or above results in a "Parameter value too large" error. Backbone objects
may be coloured using the RasMol <B>colour backbone</B> command. <P>
 The reserved
word backbone is also used as a predefined set ("help sets") and as a parameter
to the <B>set hbond</B> and <B>set ssbond</B> commands. The RasMol command <B>trace</B> renders
a smoothed backbone, in contrast to <B>backbone</B> which connects alpha carbons
with straight lines. <P>
 The backbone may be displayed with dashed lines by
use of the <B>backbone dash</B> command. <P>
 </DD>

<DT><B>Background</B> </DT>
<DD>The RasMol <B>background</B> command
is used to set the colour of the "canvas" background. The colour may be
given as either a colour name or a comma separated triple of Red, Green
and Blue (RGB) components enclosed in square brackets. Typing the command
<B>help colours</B> will give a list of the predefined colour names recognised
by RasMol. When running under X Windows, RasMol also recognises colours
in the X server's colour name database. <P>
 The <B>background</B> command is synonymous
with the RasMol <B>set background</B> command. <P>
 </DD>

<DT><B>Bond</B> </DT>
<DD>The RasMol command <B>bond &lt;number&gt;
&lt;number&gt; +</B> adds the designated bond to the drawing, increasing the bond order
if the bond already exists.  The command <B>bond &lt;number&gt; &lt;number&gt; pick</B> selects
the two atoms specified by the atom serial numbers as the two ends of a
bond around which the <B>rotate bond &lt;angle&gt;</B> command will be applied.  If no
bond exists, it is created. <P>
 Rotation around a previously picked bond may
be specified by the <B>rotate bond &lt;angle&gt;</B> command, or may also be controlled
with the mouse, using the <B>bond rotate on/off</B> or the equivalent <B>rotate bond
on/off</B> commands. <P>
 </DD>

<DT><B>Cartoon</B> </DT>
<DD>The RasMol <B>cartoon</B> command does a display of a
molecule <B>ribbons</B> as Richardson (MolScript) style protein <B>cartoons,</B> implemented
as thick (deep) ribbons. The easiest way to obtain a cartoon representation
of a protein is to use the <B>Cartoons</B> option on the <B>Display</B> menu. The <B>cartoon</B>
command represents the currently selected residues as a deep ribbon with
width specified by the command's argument. Using the command without a parameter
results in the ribbon's width being taken from the protein's secondary structure,
as described in the <B>ribbons</B> command. By default, the C-termini of beta-sheets
are displayed as arrow heads. This may be enabled and disabled using the
<B>set cartoons</B> command. The depth of the cartoon may be adjusted using the
<B>set cartoons &lt;number&gt;</B> command. The <B>set cartoons</B> command without any parameters
returns these two options to their default values. <P>
 </DD>

<DT><B>Centre</B> </DT>
<DD>The RasMol <B>centre</B>
command defines the point about which the <B>rotate</B> command and the scroll
bars rotate the current molecule. Without a parameter the centre command
resets the centre of rotation to be the centre of gravity of the molecule.
If an atom expression is specified, RasMol rotates the molecule about the
centre of gravity of the set of atoms specified by the expression. Hence,
if a single atom is specified by the expression, that atom will remain
'stationary' during rotations. <P>
 Type <B>help expression</B> for more information
on RasMol atom expressions. <P>
 Alternatively the centring may be given as
a comma separated triple of [CenX, CenY, CenZ] offsets in RasMol units
(1/250 of an Angstrom) from the centre of gravity.  The triple must be enclosed
in square brackets. <P>
 The optional forms <B>centre ... translate</B> and <B>centre ... center</B>
may be used to specify use of a translated centre of rotation (not necessarily
in the centre of the canvas) or a centre of rotation which is placed at
the centre of the canvas.  Starting with RasMol 2.7.2, the default is to center
the new axis on the canvas. <P>
 </DD>

<DT><B>Clipboard</B> </DT>
<DD>The RasMol <B>clipboard</B> command places
a copy of the currently displayed image on the local graphics 'clipboard'.
Note: this command is not yet supported on UNIX or VMS machines. It is intended
to make transfering images between applications easier under Microsoft
Windows or on an Apple Macintosh. <P>
 When using RasMol on a UNIX or VMS system
this functionality may be achieved by generating a raster image in a format
that can be read by the receiving program using the RasMol <B>write</B> command.
<P>
 </DD>

<DT><B>Colour</B> </DT>
<DD>Colour the atoms (or other objects) of the selected region. The
colour may be given as either a colour name or a comma separated triple
of Red, Green and Blue (RGB) components enclosed in square brackets. Typing
the command <B>help colours</B> will give a list of all the predefined colour
names recognised by RasMol. <P>
 Allowed objects are <B>atoms,</B> <B>bonds,</B> <B>backbone,</B>
<B>ribbons,</B> <B>labels,</B> <B>dots,</B> <B>hbonds</B> and <B>ssbonds.</B> If no object is specified, the
default keyword <B>atom</B> is assumed. Some colour schemes are defined for certain
object types. The colour scheme <B>none</B> can be applied to all objects except
atoms and dots, stating that the selected objects have no colour of their
own, but use the colour of their associated atoms (i.e. the atoms they connect).
<B>Atom</B> objects can also be coloured by <B>alt,</B> <B>amino,</B> <B>chain,</B> <B>charge,</B> <B>cpk,</B> <B>group,</B>
<B>model,</B> <B>shapely,</B> <B>structure,</B> <B>temperature</B> or <B>user.</B> Hydrogen bonds can also
be coloured by <B>type</B> and dot surfaces can also be coloured by <B>electrostatic
potential.</B> For more information type <B>help colour &lt;colour&gt;.</B> <P>
 </DD>

<DT><B>Connect</B> </DT>
<DD>The RasMol
<B>connect</B> command is used to force RasMol to (re)calculate the connectivity
of the current molecule. If the original input file contained connectivity
information, this is discarded. The command <B>connect false</B> uses a fast heuristic
algorithm that is suitable for determining bonding in large bio-molecules
such as proteins and nucleic acids. The command <B>connect true</B> uses a slower
more accurate algorithm based upon covalent radii that is more suitable
to small molecules containing inorganic elements or strained rings. If no
parameters are given, RasMol determines which algorithm to use based on
the number of atoms in the input file. Greater than 255 atoms causes RasMol
to use the faster implementation. This is the method used to determine bonding,
if necessary, when a molecule is first read in using the <B>load</B> command. <P>

</DD>

<DT><B>Define</B> </DT>
<DD>The RasMol <B>define</B> command allows the user to associate an arbitrary
set of atoms with a unique identifier. This allows the definition of user-defined
sets. These sets are declared statically, i.e. once defined the contents of
the set do not change, even if the expression defining them depends on
the current transformation and representation of the molecule. <P>
 </DD>

<DT><B>Depth</B> </DT>
<DD>The
RasMol <B>depth</B> command enables, disables or positions the back-clipping plane
of the molecule. The program only draws those portions of the molecule that
are closer to the viewer than the clipping plane. Integer values range from
zero at the very back of the molecule to 100 which is completely in front
of the molecule. Intermediate values determine the percentage of the molecule
to be drawn. <P>
 This command interacts with the <B>slab &lt;value&gt;</B> command, which
clips to the front of a given z-clipping plane. <P>
 </DD>

<DT><B>Dots</B> </DT>
<DD>The RasMol <B>dots</B> command
is used to generate a van der Waals' dot surface around the currently selected
atoms. Dot surfaces display regularly spaced points on a sphere of van der
Waals' radius about each selected atom. Dots that would are 'buried' within
the van der Waals' radius of any other atom (selected or not) are not displayed.
The command <B>dots on</B> deletes any existing dot surface and generates a dots
surface around the currently selected atom set with a default dot density
of 100. The command <B>dots off</B> deletes any existing dot surface. The dot density
may be specified by providing a numeric parameter between 1 and 1000. This
value approximately corresponds to the number of dots on the surface of
a medium sized atom. <P>
 By default, the colour of each point on a dot surface
is the colour of its closest atom at the time the surface is generated.
The colour of the whole dot surface may be changed using the <B>colour dots</B>
command. <P>
 </DD>

<DT><B>Echo</B> </DT>
<DD>The RasMol <B>echo</B> command is used to display a message in the
RasMol command/terminal window. The string parameter may optionally be delimited
in double quote characters. If no parameter is specified, the <B>echo</B> command
displays a blank line. This command is particularly useful for displaying
text from within a RasMol <B>script</B> file. <P>
 </DD>

<DT><B>English</B> </DT>
<DD>The RasMol <B>English</B> command
sets the menus and messages to the English versions. The commands <B>French,</B>
<B>Italian</B> and <B>Spanish</B> may be used to select French, Italian and Spanish menus
and messages. <P>
 </DD>

<DT><B>French</B> </DT>
<DD>The RasMol <B>French</B> command sets the menus and messages
to the French versions. The commands <B>English,</B> <B>Italian</B> and <B>Spanish</B> may be
used to select English, Italian and Spanish menus and messages. <P>
 </DD>

<DT><B>HBonds</B>
</DT>
<DD>The RasMol <B>hbond</B> command is used to represent the hydrogen bonding of the
protein molecule's backbone. This information is useful in assessing the
protein's secondary structure. Hydrogen bonds are represented as either dotted
lines or cylinders between the donor and acceptor residues. The first time
the <B>hbond</B> command is used, the program searches the structure of the molecule
to find hydrogen bonded residues and reports the number of bonds to the
user. The command <B>hbonds on</B> displays the selected 'bonds' as dotted lines,
and the <B>hbonds off</B> turns off their display. The colour of hbond objects
may be changed by the <B>colour hbond</B> command. Initially, each hydrogen bond
has the colours of its connected atoms. <P>
 By default the dotted lines are
drawn between the accepting oxygen and the donating nitrogen. By using the
<B>set hbonds</B> command the alpha carbon positions of the appropriate residues
may be used instead. This is especially useful when examining proteins in
backbone representation. <P>
 </DD>

<DT><B>Help</B> </DT>
<DD>The RasMol <B>help</B> command provides on-line help
on the given topic. <P>
 </DD>

<DT><B>Italian</B> </DT>
<DD>The RasMol <B>Italian</B> command sets the menus and
messages to the Italian versions. The commands <B>English,</B> <B>French</B> and <B>Spanish</B>
may be used to select English, French and Spanish menus and messages. <P>
 </DD>

<DT><B>Label</B>
</DT>
<DD>The RasMol <B>label</B> command allows an arbitrary formatted text string to be
associated with each currently selected atom.  This string may contain embedded
'expansion specifiers' which display properties of the atom being labelled.
An expansion specifier consists of a '%' character followed by a single alphabetic
character specifying the property to be displayed. An actual '%' character
may be displayed by using the expansion specifier '%%'. <P>
 Atom labelling for
the currently selected atoms may be turned off with the command <B>label off.</B>
By default, if no string is given as a parameter, RasMol uses labels appropriate
for the current molecule. <P>
 The colour of each label may be changed using
the <B>colour label</B> command. By default, each label is drawn in the same colour
as the atom to which it is attached. The size and spacing of the displayed
text may be changed using the <B>set fontsize</B> command.  The width of the strokes
in the displayed text may be changed  using the<BR>
 <B>set fontstroke</B>  command.<BR>
 <P>
 </DD>

<DT><B>Load</B> </DT>
<DD>Load a molecule coordinate file into RasMol. Valid molecule file
formats are <B>pdb</B> (Protein Data Bank format), <B>mdl</B> (Molecular Design Limited's
MOL file format), <B>alchemy</B> (Tripos' Alchemy file format), <B>mol2</B> (Tripos' Sybyl
Mol2 file format), <B>charmm</B> (CHARMm file format), <B>xyz</B> (MSC's XMol XYZ file
format), <B>mopac</B> (J. P. Stewart's MOPAC file format) or <B>cif</B> (IUCr CIF or mmCIF
file format). If no file format is specified, <B>PDB,</B> <B>CIF,</B> or <B>mmCIF</B> is assumed
by default. Up to 5 molecules may be loaded at a time. To delete a molecule
prior to loading another use the RasMol <B>zap</B> command.  To select a molecule
for manipulation use the RasMol <B>molecule &lt;n&gt;</B> command. <P>
 The <B>load</B> command selects
all the atoms in the molecule, centres it on the screen and renders it
as a CPK coloured wireframe model. If the molecule contains no bonds (i.e.
contains only alpha carbons), it is drawn as an alpha carbon backbone. If
the file specifies fewer bonds than atoms, RasMol determines connectivity
using the <B>connect</B> command. <P>
 The <B>load inline</B> command also allows the storing
of atom coordinates in scripts to allow better integration with WWW browsers.
A load command executed inside a script file may specify the keyword <B>inline</B>
instead of a conventional filename. This option specifies that the coordinates
of the molecule to load are stored in the same file as the currently executing
commands. <P>
 </DD>

<DT><B>Molecule</B> </DT>
<DD>The RasMol <B>molecule</B> command selects one of up to 5 previously
loaded molecules for active manipulation.  While all the molcules are displayed
and may be rotated collectively (see the <B>rotate all</B> command), only one
molecule at a time time is active for manipulation by the commands which
control the details of rendering. <P>
 </DD>

<DT><B>Monitor</B> </DT>
<DD>The RasMol <B>monitor</B> command allows
the display of distance monitors. A distance monitor is a dashed (dotted)
line between an arbitrary pair of atoms, optionally labelled by the distance
between them. The RasMol command <B>monitor &lt;number&gt; &lt;number&gt;</B> adds such a distance
<P>
monitor between the two atoms specified by the atom serial numbers given
<P>
as parameters <P>
 Distance monitors are turned off with the command <B>monitors
off.</B> By default, monitors display the distance between its two end points
as a label at the centre of the monitor. These distance labels may be turned
off with the command <B>set monitors off,</B> and re-enabled with the command <B>set
monitors on.</B> Like most other representations, the colour of a monitor is
taken from the colour of its end points unless specified by the <B>colour
monitors</B> command. <P>
 Distance monitors may also be added to a molecule interactively
with the mouse, using the <B>set picking monitor</B> command. Clicking on an atom
results in its being identified on the rasmol command line. In addition
every atom picked increments a modulo counter such that, in monitor mode,
every second atom displays the distance between this atom and the previous
one. The shift key may be used to form distance monitors between a fixed
atom and several consecutive positions. A distance monitor may also be removed
(toggled) by selecting the appropriate pair of atom end points a second
time. <P>
 </DD>

<DT><B>Pause</B> </DT>
<DD>The RasMol <B>pause</B> command is used in script files to stop the
script file for local manipulation by a mouse, until any key is pushed
to restart the script file. <B>Wait</B> is synonymous with <B>pause.</B> This command may
be executed in RasMol script files to suspend the sequential execution
of commands and allow the user to examine the current image.   When RasMol
executes a <B>pause</B> command in a script file, it suspends  execution of the
rest of the file, refreshes the image on the screen and allows the manipulation
of the image using the mouse and scroll  bars, or resizing of the graphics
window.  Once a key is pressed, control returns to the script file at the
line following the <B>pause</B> command.  While a script is suspended the molecule
may be rotated, translated, scaled, slabbed and picked as usual, but all
menu commands are disabled. <P>
 </DD>

<DT><B>Print</B> </DT>
<DD>The RasMol <B>print</B> command sends the currently
displayed image to the local default printer using the operating system's
native printer driver. Note: this command is not yet supported under UNIX
or VMS. It is intended to take advantage of Microsoft Windows and Apple
Macintosh printer drivers. For example, this allows images to be printed
directly on a dot matrix printer. <P>
 When using RasMol on a UNIX or VMS system
this functionality may be achieved by either generating a PostScript file
using the RasMol <B>write ps</B> or <B>write vectps</B> commands and printing that or
generating a raster image file and using a utility to dump that to the
local printer. <P>
 </DD>

<DT><B>Quit</B> </DT>
<DD>Exit from the RasMol program. The RasMol commands <B>exit</B>
and <B>quit</B> are synonymous, except within nested scripts.  In that case, <B>exit</B>
terminates only the current level, while <B>quit</B> terminates all nested levels
of scripts. <P>
 </DD>

<DT><B>Refresh</B> </DT>
<DD>The RasMol <B>refresh</B> command redraws the current image.
 This is useful in scripts to ensure application of a complex list of parameter
changes. <P>
 </DD>

<DT><B>Renumber</B> </DT>
<DD>The RasMol <B>renumber</B> command sequentially numbers the
residues in a macromolecular chain. The optional parameter specifies the
value of the first residue in the sequence. By default, this value is one.
For proteins, each amino acid is numbered consecutively from the N terminus
to the C terminus. For nucleic acids, each base is numbered from the 5' terminus
to the 3' terminus. All chains in the current database are renumbered and
gaps in the original sequence are ignored. The starting value for numbering
may be negative. <P>
 </DD>

<DT><B>Reset</B> </DT>
<DD>The RasMol <B>reset</B> command restores the original viewing
transformation and centre of rotation. The scale is set to its default value,
<B>zoom 100,</B> the centre of rotation is set to the geometric centre of the
currently loaded molecule, <B>centre all,</B> this centre is translated to the
middle of the screen and the viewpoint set to the default orientation. <P>

This command should not be mistaken for the RasMol <B>zap</B> command which deletes
the currently stored molecule, returning the program to its initial state.
<P>
 </DD>

<DT><B>Restrict</B> </DT>
<DD>The RasMol <B>restrict</B> command both defines the currently selected
region of the molecule and disables the representation of (most of) those
parts of the molecule no longer selected.  All subsequent RasMol commands
that modify a molecule's colour or representation affect only the currently
selected region. The parameter of a <B>restrict</B> command is a RasMol atom expression
that is evaluated for every atom of the current molecule. This command is
very similar to the RasMol <B>select</B> command, except <B>restrict</B> disables the
<B>wireframe,</B> <B>spacefill</B> and <B>backbone</B> representations in the non-selected region.
<P>
 Type "help expression" for more information on RasMol atom expressions.
<P>
 </DD>

<DT><B>Ribbons</B> </DT>
<DD>The RasMol <B>ribbons</B> command displays the currently loaded protein
or nucleic acid as a smooth solid "ribbon" surface passing along the backbone
of the protein. The ribbon is drawn between each amino acid whose alpha
carbon is currently selected. The colour of the ribbon is changed by the
RasMol <B>colour ribbon</B> command. If the current ribbon colour is <B>none</B> (the
default), the colour is taken from the alpha carbon at each position along
its length. <P>
 The width of the ribbon at each position is determined by the
optional parameter in the usual RasMol units. By default the width of the
ribbon is taken from the secondary structure of the protein or a constant
value of 720 (2.88 Angstroms) for nucleic acids. The default width of protein
alpha helices and beta sheets is 380 (1.52 Angstroms) and 100 (0.4 Angstroms)
for turns and random coil. The secondary structure assignment is either
from the PDB file or calculated using the DSSP algorithm as used by the
<B>structure</B> command. This command is similar to the RasMol command <B>strands</B>
which renders the biomolecular ribbon as parallel depth-cued curves. <P>
 </DD>

<DT><B>Rotate</B>
</DT>
<DD>Rotate the molecule about the specified axis. Permitted values for the axis
parameter are "x", "y", "z" and "bond". The integer parameter states the
angle in degrees for the structure to be rotated. For the X and Y axes,
positive values move the closest point up and right, and negative values
move it down and left, respectively. For the Z axis, a positive rotation
acts clockwise and a negative angle anti-clockwise. <P>
 Alternatively, this
command may be used to specify which rotations the mouse or dials will
control.  If <B>rotate bond true</B> is selected, the horizontal scroll bar will
control rotation around the axis selected by the <B>bond src dst pick</B> command.
 If <B>rotate all true</B> is selected, and multiple molecules have been loaded,
then all molecules will rotate together.  In all other cases, the mouseand
dials control the the rotation of the molecule selected by the <B>molecule
n</B> command. <P>
 </DD>

<DT><B>Save</B> </DT>
<DD>Save the currently selected set of atoms in a Protein Data
Bank (PDB), MDL, Alchemy(tm) or XYZ format file. The distinction between
this command and the RasMol <B>write</B> command has been dropped. The only difference
is that without a format specifier the <B>save</B> command generates a <B>PDB</B> file
and the <B>write</B> command generates a <B>GIF</B> image. <P>
 </DD>

<DT><B>Script</B> </DT>
<DD>The RasMol <B>script</B> command
reads a set of RasMol commands sequentially from a text file and executes
them. This allows sequences of commonly used commands to be stored and performed
by single command. A RasMol script file may contain a further script command
up to a maximum "depth" of 10, allowing complicated sequences of actions
to be executed. RasMol ignores all characters after the first '#' character
on each line allowing the scripts to be annotated. Script files are often
also annotated using the RasMol <B>echo</B> command. <P>
 The most common way to generate
a RasMol script file is to use the <B>write script</B> or <B>write rasmol</B> commands
to output the sequence of commands that are needed to regenerate the current
view, representation and colouring of the currently displayed molecule.
<P>
 The RasMol command <B>source</B> is synonymous with the <B>script</B> command. <P>
 </DD>

<DT><B>Select</B>
</DT>
<DD>Define the currently selected region of the molecule. All subsequent RasMol
commands that manipulate a molecule or modify its colour or representation
only affect the currently selected region. The parameter of a <B>select</B> command
is a RasMol expression that is evaluated for every atom of the current
molecule. The currently selected (active) region of the molecule are those
atoms that cause the expression to evaluate true. To select the whole molecule
use the RasMol command <B>select all.</B> The behaviour of the <B>select</B> command without
any parameters is determined by the RasMol <B>hetero</B> and <B>hydrogen</B> parameters.
<P>
 Type "help expression" for more information on RasMol atom expressions.
<P>
 </DD>

<DT><B>Set</B> </DT>
<DD>The RasMol <B>set</B> command allows the user to alter various internal program
parameters such as those controlling rendering options. Each parameter has
its own set or permissible parameter options. Typically, omitting the paramter
option resets that parameter to its default value. A list of valid parameter
names is given below. <P>
 </DD>

<DT><B>Show</B> </DT>
<DD>The RasMol <B>show</B> command display details of the
status of the currently loaded molecule. The command <B>show information</B> lists
the molecule's name, classification, PDB code and the number of atoms, chains,
groups it contains. If hydrogen bonding, disulphide bridges or secondary
structure have been determined, the number of hbonds, ssbonds, helices,
ladders and turns are also displayed, respectively. The command <B>show centre</B>
shows any non-zero centering values selected by the <B>centre [CenX, CenY,
CenZ]</B> command. The command <B>show phipsi</B> shows the phi and psi angles of the
currently selected residues and the omega angles of cis peptide bonds. The
command <B>show RamPrint</B> (or 'show RPP' or 'show RamachandranPrinterPlot') shows
a simple Ramachandran printer plot in the style of Frances Bernstein's fisipl
program.  The command <B>show rotation</B> (or 'show rot' or 'show 'rotate') shows the
currently selected values of z, y, x and bond rotations, if any. The command
<B>show selected</B> (or 'show selected group' or 'show selected chain' or  'show selected
atom' ) shows the groups (default), chains or atoms of the current selection.
The command <B>show sequence</B> lists the residues that comprise each chain of
the molecule.  The command <B>show symmetry</B> shows the space group and unit
cell of the molecule. The command <B>show translation</B> shows any non-zero translation
values selected by the <B>translate &lt;axis&gt; &lt;value&gt;</B> command. The command <B>show zoom</B>
shows any non-zero zoom value selected by the <B>zoom &lt;value&gt;</B> command. <P>
 </DD>

<DT><B>Slab</B> </DT>
<DD>The
RasMol <B>slab</B> command enables, disables or positions the z-clipping plane
of the molecule. The program only draws those portions of the molecule that
are further from the viewer than the slabbing plane. Integer values range
from zero at the very back of the molecule to 100 which is completely in
front of the molecule. Intermediate values determine the percentage of the
molecule to be drawn. <P>
 This command interacts with the <B>depth &lt;value&gt;</B> command,
which clips to the rear of a given z-clipping plane. <P>
 </DD>

<DT><B>Spacefill</B> </DT>
<DD>The RasMol
<B>spacefill</B> command is used to represent all of the currently selected atoms
as solid spheres. This command is used to produce both union-of-spheres and
ball-and-stick models of a molecule. The command, <B>spacefill true,</B> the default,
represents each atom as a sphere of van der Waals radius. The command <B>spacefill
off</B> turns off the representation of the selected atom as spheres. A sphere
radius may be specified as an integer in RasMol units (1/250th Angstrom)
or a value containing a decimal point. A value of 500 (2.0 Angstroms) or
greater results in a "Parameter value too large" error. <P>
 The <B>temperature</B>
option sets the radius of each sphere to the value stored in its temperature
field. Zero or negative values have no effect and values greater than 2.0
are truncated to 2.0.  The <B>user</B> option allows the radius of each sphere to
be specified by additional lines in the molecule's PDB file using Raster
3D's COLOUR record extension. <P>
 The RasMol command <B>cpk</B> is synonymous with
the <B>spacefill</B> command. <P>
 </DD>

<DT><B>Spanish</B> </DT>
<DD>The RasMol <B>Spanish</B> command sets the menus
and messages to the Spanish versions. The commands <B>English,</B> <B>French</B> and <B>Italian</B>
may be used to select English, French and Italian menus and messages. <P>
 </DD>

<DT><B>SSBonds</B>
</DT>
<DD>The RasMol <B>ssbonds</B> command is used to represent the disulphide bridges
of the protein molecule as either dotted lines or cylinders between the
connected cysteines. The first time that the <B>ssbonds</B> command is used, the
program searches the structure of the protein to find half-cysteine pairs
(cysteines whose sulphurs are within 3 Angstroms of each other) and reports
the number of bridges to the user. The command <B>ssbonds on</B> displays the selected
"bonds" as dotted lines, and the command <B>ssbonds off</B> disables the display
of ssbonds in the currently selected area. Selection of disulphide bridges
is identical to normal bonds, and may be adjusted using the RasMol <B>set
bondmode</B> command. The colour of disulphide bonds may be changed using the
<B>colour ssbonds</B> command. By default, each disulphide bond has the colours
of its connected atoms. <P>
 By default disulphide bonds are drawn between the
sulphur atoms within the cysteine groups. By using the <B>set ssbonds</B> command
the position of the cysteine's alpha carbons may be used instead. <P>
 </DD>

<DT><B>Star</B> </DT>
<DD>The
RasMol <B>star</B> command is used to represent all of the currently selected
atoms as stars (six strokes, one each in the x, -x, y, -y, z and -z directions).
The commands <B>select not bonded</B> followed by <B>star 75</B> are useful to mark unbonded
atoms in a <B>wireframe</B> display with less overhead than provided by <B>spacefill
75.</B> This can be done automatically for all subsequent wireframe displays
with the command <B>set bondmode not bonded.</B> <P>
 The command <B>star true,</B> the default,
represents each atom as a star with strokes length equal to van der Waals
radius. The command <B>star off</B> turns off the representation of the selected
atom as stars. A star stroke length may be specified as an integer in RasMol
units (1/250th Angstrom) or a value containing a decimal point. A value
of 500 (2.0 Angstroms) or greater results in a "Parameter value too large"
error. <P>
 The <B>temperature</B> option sets the stroke length of each star to the
value stored in its temperature field. Zero or negative values have no effect
and values greater than 2.0 are truncated to 2.0.  The <B>user</B> option allows
the stroke length of each star to be specified by additional lines in the
molecule's PDB file using Raster 3D's COLOUR record extension. <P>
 The RasMol
<B>spacefill</B> command can be used for more artistic rendering of atoms as spheres.
<P>
 </DD>

<DT><B>Stereo</B> </DT>
<DD>The RasMol <B>stereo</B> command provides side-by-side stereo display of
images. Stereo viewing of a molecule may be turned on (and off) either by
selecting <B>Stereo</B> from the <B>Options</B> menu, or by typing the commands <B>stereo
on</B> or <B>stereo off.</B> <P>
 Starting with RasMol version 2.7.2.1, the <B>Stereo</B> menu selection
and the command <B>stereo</B> without arguments cycle from the initial state of
<B>stereo off</B> to <B>stereo on</B> in cross-eyed mode to <B>stereo on</B> in wall-eyed mode
and then back to <B>stereo off.</B> <P>
 The separation angle between the two views
may be adjusted with the <B>set stereo [-] &lt;number&gt;</B> command, where positive values
result in crossed eye viewing and negative values in relaxed (wall-eyed)
viewing. The inclusion of <B>[-] &lt;number&gt;</B> in the <B>stereo</B> command, as for example
in <B>stereo 3</B> or <B>stereo -5,</B> also controls angle and direction. <P>
 The stereo
command is only partially implemented. When stereo is turned on, the image
is not properly recentred. (This can be done with a <B>translate x -&lt;number&gt;</B> 
command.)<BR>
 It is not supported in vector PostScript output files, is not saved by
the <B>write script</B> command, and in general is not yet properly interfaced
with several other features of the program. <P>
 </DD>

<DT><B>Strands</B> </DT>
<DD>The RasMol <B>strands</B>
command displays the currently loaded protein or nucleic acid as a smooth
"ribbon" of depth-cued curves passing along the backbone of the protein.
The ribbon is composed of a number of strands that run parallel to one
another along the peptide plane of each residue. The ribbon is drawn between
each amino acid whose alpha carbon is currently selected. The colour of
the ribbon is changed by the RasMol <B>colour ribbon</B> command. If the current
ribbon colour is <B>none</B> (the default), the colour is taken from the alpha
carbon at each position along its length. The central and outermost strands
may be coloured independently using the <B>colour ribbon1</B> and <B>colour ribbon2</B>
commands, respectively. The number of strands in the ribbon may be altered
using the RasMol <B>set strands</B> command. <P>
 The width of the ribbon at each position
is determined by the optional parameter in the usual RasMol units. By default
the width of the ribbon is taken from the secondary structure of the protein
or a constant value of 720 for nucleic acids (which produces a ribbon 2.88
Angstroms wide). The default width of protein alpha helices and beta sheets
is 380 (1.52 Angstroms) and 100 (0.4 Angstroms) for turns and random coil.
The secondary structure assignment is either from the PDB file or calculated
using the DSSP algorithm as used by the <B>structure</B> command. This command
is similar to the RasMol command <B>ribbons</B> which renders the biomolecular
ribbon as a smooth shaded surface. <P>
 </DD>

<DT><B>Structure</B> </DT>
<DD>The RasMol <B>structure</B> command
calculates secondary structure assignments for the currently loaded protein.
If the original PDB file contained structural assignment records (HELIX,
SHEET and TURN) these are discarded. Initially, the hydrogen bonds of the
current molecule are found, if this hasn't been done already. The secondary
structure is then determined using Kabsch and Sander's DSSP algorithm. Once
finished the program reports the number of helices, strands and turns found.
<P>
 </DD>

<DT><B>Trace</B> </DT>
<DD>The RasMol <B>trace</B> command displays a smooth spline between  consecutive
alpha carbon positions.  This spline does not pass exactly through the alpha
carbon position of each residue, but  follows the same path as <B>ribbons,</B>
<B>strands</B> and <B>cartoons.</B> Note that residues may be displayed as <B>ribbons,</B> <B>strands,</B>
<B>cartoons</B> or as a <B>trace.</B> Enabling one of these representations disables the
others. However, a residue may be displayed simultaneously as backbone and
as one of the above representations.  This may change in future versions
of RasMol.  Prior to version 2.6, <B>trace</B> was synonymous with <B>backbone.</B> <P>
 <B>Trace
temperature</B> displays the backbone as a wider cylinder at high temperature
factors and thinner at lower.  This representation is useful to X-ray crystallographers
and NMR spectroscopists. <P>
 </DD>

<DT><B>Translate</B> </DT>
<DD>The RasMol <B>translate</B> command moves the
position of the centre of the molecule on the screen. The axis parameter
specifies along which axis the molecule is to be moved and the integer
parameter specifies the absolute position of the molecule centre from the
middle of the screen. Permitted values for the axis parameter are "x", "y"
and "z". Displacement values must be between -100 and 100 which correspond
to moving the current molecule just off the screen. A positive "x" displacement
moves the molecule to the right, and a positive "y" displacement moves
the molecule down the screen. The pair of commands <B>translate x 0</B> and <B>translate
y 0</B> centres the molecule on the screen. <P>
 </DD>

<DT><B>UnBond</B> </DT>
<DD>The RasMol command <B>unbond
&lt;number&gt; &lt;number&gt;</B> removes the designated bond from the drawing. <P>
 The command
<B>unbond</B> without arguments removes a bond previously picked by the <B>bond &lt;number&gt;
&lt;number&gt; pick</B> command. <P>
 </DD>

<DT><B>Wireframe</B> </DT>
<DD>The RasMol <B>wireframe</B> command represents
each bond within the selected region of the molecule as a cylinder, a line
or a depth-cued vector. The display of bonds as depth-cued vectors (drawn
darker the further away from the viewer) is turned on by the command <B>wireframe</B>
or <B>wireframe on.</B> The selected bonds are displayed as cylinders by specifying
a radius either as an integer in RasMol units or containing a decimal point
as a value in Angstroms.  A parameter value of 500 (2.0 Angstroms) or above
results in an "Parameter value too large" error. Bonds may be coloured using
the <B>colour bonds</B> command. <P>
 If the selected bonds involved atoms of alternate
conformers then the bonds are narrowed in the middle to a radius of .8 of
the specified radius (or to the radius specifed as the optional second
parameter). <P>
 Non-bonded atoms, which could become invisible in an ordinary
<B>wireframe</B> display can be marked by a preceding <B>set bondmode not bonded</B>
command.  If nearly co-linear bonds to atoms cause them to be difficult to
see in a wireframe display, the <B>set bondmode all</B> command will add markers
for <B>all</B> atoms in subsequent <B>wireframe</B> command executions. <P>
 </DD>

<DT><B>Write</B> </DT>
<DD>Write the
current image to a file in a standard format. Currently supported image
file formats include <B>bmp</B> (Microsoft bitmap) and <B>gif</B> (Compuserve GIF), <B>iris</B>
(IRIS RGB), <B>ppm</B> (Portable Pixmap), <B>ras</B> (Sun rasterfile), <B>ps</B> and <B>epsf</B> (Encapsulated
PostScript), <B>monops</B> (Monochrome Encapsulated PostScript), <B>pict</B> (Apple PICT),
<B>vectps</B> (Vector Postscript).  The <B>write</B> command may also be used to generate
command scripts for other graphics programs. The format <B>script</B> writes out
a file containing the RasMol <B>script</B> commands to reproduce the current image.
The format <B>molscript</B> writes out the commands required to render the current
view of the molecule as ribbons in Per Kraulis' Molscript program and the
format <B>kinemage</B> the commands for David Richardson's program Mage.  The following
formats are useful for further processing: <B>povray</B> (POVRay 2), <B>povray3</B> (POVRay
3 -- under development), <B>vrml</B> (VRML file). Finally, several formats are provided
to provide phi-psi data for listing or for <B>phipsi</B> (phi-psi data as an annotated
list with cis omegas), <B>ramachan</B> and <B>RDF</B> and <B>RamachandranDataFile</B> (phi-psi
data as columns of numbers for gnuplot), <B>RPP</B> and <B>RamachandranPrinterPlot</B>
(phi-psi data as a printer plot). <P>
 The distinction between this command and
the RasMol <B>save</B> command has been dropped. The only difference is that without
a format specifier the <B>save</B> command generates a <B>PDB</B> file and the <B>write</B>
command generates a <B>GIF</B> image. <P>
 </DD>

<DT><B>Zap</B> </DT>
<DD>Deletes the contents of the current
database and resets parameter variables to their initial default state.
<P>
 </DD>

<DT><B>Zoom</B> </DT>
<DD>Change the magnification of the currently displayed image. Boolean
parameters either magnify or reset the scale of current molecule. An integer
parameter specifies the desired magnification as a percentage of the default
scale. The minimum parameter value is 10; the maximum parameter value is
dependent upon the size of the molecule being displayed. For medium sized
proteins this is about 500. <P>
 </DD>
</DL>

<H2><A NAME="sect6" HREF="#toc6">Set Parameters</A></H2>
RasMol has a number of internal
parameters that may be modified using the <B>set</B> command. These parameters
control a number of program options such as rendering options and mouse
button mappings. <P>
 
<DL>

<DT><B>Set Ambient</B> </DT>
<DD>The RasMol <B>ambient</B> parameter is used to control
the amount of ambient (or surrounding) light in the scene. The <B>ambient</B> value
must be between 0 and 100. It controls the percentage intensity of the darkest
shade of an object. For a solid object, this is the intensity of surfaces
facing away from the light source or in shadow. For depth-cued objects this
is the intensity of objects furthest from the viewer. <P>
 This parameter is
commonly used to correct for monitors with different "gamma values" (brightness),
to change how light or dark a hardcopy image appears when printed or to
alter the feeling of depth for wireframe or ribbon representations. <P>
 </DD>

<DT><B>Set
Axes</B> </DT>
<DD>The RasMol <B>axes</B> parameter controls the display of orthogonal coordinate
axes on the current display. The coordinate axes are those used in the molecule
data file, and the origin is the centre of the molecule's bounding box. The
<B>set axes</B> command is similar to the commands <B>set boundbox</B> and <B>set unitcell</B>
that display the bounding box and the crystallographic unit cell, respectively.
<P>
 </DD>

<DT><B>Set Backfade</B> </DT>
<DD>The RasMol <B>backfade</B> parameter is used to control backfade
to the specified background colour, rather than black.  This is controlled
by the commands <B>set backfade on</B> and <B>set backfade off.</B> For example, this
may be used to generate depth-cued images that fade to white, rather than
black. <P>
 </DD>

<DT><B>Set Background</B> </DT>
<DD>The RasMol <B>background</B> parameter is used to set the
colour of the "canvas" background. The colour may be given as either a colour
name or a comma separated triple of Red, Green, Blue (RGB) components enclosed
in square brackets. Typing the command <B>help colours</B> will give a list of
the predefined colour names recognised by RasMol. When running under X Windows,
RasMol also recognises colours in the X server's colour name database. <P>
 The
command <B>set background</B> is synonymous with the RasMol command <B>background.</B>
<P>
 </DD>

<DT><B>Set BondMode</B> </DT>
<DD>The RasMol <B>set bondmode</B> command controls the mechanism used
to select individual bonds and modifies the display of bonded and non-bonded
atoms by subsequent <B>wireframe</B> commands. <P>
 When using the <B>select</B> and <B>restrict</B>
commands, a given bond will be selected if i) the bondmode is <B>or</B> and either
of the connected atoms is selected, or ii) the bondmode is <B>and</B> and both
atoms connected by the bond are selected. Hence an individual bond may be
uniquely identified by using the command <B>set bondmode and</B> and then uniquely
selecting the atoms at both ends. <P>
 The <B>bondmode [all | none | not bonded]</B>
commands add <B>star 75</B> or <B>spacefill 75</B> markers for the designated atoms to
<B>wireframe</B> displays.  Stars are used when the specified wireframe radius
is zero. <P>
 </DD>

<DT><B>Set Bonds</B> </DT>
<DD>The RasMol <B>bonds</B> parameter is used to control display
of double and triple bonds as multiple  lines or cylinders.  Currently bond
orders are only read from  MDL Mol files, Sybyl Mol2 format files, Tripos
Alchemy format files, CIF and mmCIF,  and suitable PDB files.  Double (and
triple)  bonds are specified in some  PDB files by specifying a given bond
twice  (and three times) in CONECT records.  The  command <B>set bonds on</B> enables
the display of bond orders, and  the command <B>set bonds off</B> disables them.
<P>
 </DD>

<DT><B>Set BoundBox</B> </DT>
<DD>The RasMol <B>boundbox</B> parameter controls the display of the
current molecule's bounding box on the display. The bounding box is orthogonal
to the data file's original coordinate axes. The <B>set boundbox</B> command is
similar to the commands <B>set axes</B> and <B>set unitcell</B> that display orthogonal
coordinate axes and the bounding box, respectively. <P>
 </DD>

<DT><B>Set Cartoon</B> </DT>
<DD>The RasMol
<B>cartoon</B> parameter is used to control display of the cartoon version of
the <B>ribbons</B> display.  By default, the C-termini of beta-sheets are displayed
as arrow heads. This may be enabled and disabled using the <B>set cartoons
&lt;boolean&gt;</B> command. The depth of the cartoon may be adjusted using the <B>cartoons
&lt;number&gt;</B> command. The <B>set cartoons</B> command without any parameters returns
these two options to  their default values.<BR>
 <P>
 </DD>

<DT><B>Set CisAngle</B> </DT>
<DD>The RasMol <B>cisangle</B> parameter controls the cutoff angle
for identifying cis peptide  bonds.  If no value is given, the cutoff is
set to 90 degrees.<BR>
 <P>
 </DD>

<DT><B>Set Display</B> </DT>
<DD>This command controls the display mode within RasMol. By default,
<B>set display normal,</B> RasMol displays the molecule in the representation
specified by the user. The command <B>set display selected</B> changes the display
mode such that the molecule is temporarily drawn so as to indicate currently
selected portion of the molecule. The user specified colour scheme and representation
remains unchanged. In this representation all selected atoms are shown in
yellow and all non selected atoms are shown in blue. The colour of the background
is also changed to a dark grey to indicate the change of display mode. This
command is typically only used by external Graphical User Interfaces (GUIs).
<P>
 </DD>

<DT><B>Set FontSize</B> </DT>
<DD>The RasMol <B>set fontsize</B> command is used to control the size
of the characters that form atom labels. This value corresponds to the height
of the displayed character in pixels. The maximum value of <B>fontsize</B> is 48
pixels, and the default value is 8 pixels high. Fixed or proportional spacing
may be selected by appending the "FS" or "PS" modifiers, respectively. 
The default is "FS". To display atom labels on the screen use the RasMol
<B>label</B> command and to change the colour of displayed labels, use the <B>colour
labels</B> command. <P>
 </DD>

<DT><B>Set FontStroke</B> </DT>
<DD>The RasMol <B>set fontstroke</B> command is used
to control the size of the stroke width of the characters that form atom
labels.  This value is the radius in pixels of cylinders used to form the
strokes.  The special value of "0" is the default used for the normal single
pixel stroke width, which allows for rapid drawing and rotation of the
image. Non-zero values are provided to allow for more artistic atom labels
for publication at the expense of extra time in rendering the image. <P>
 When
wider strokes are used, a larger font size is recommend, e.g. by using the
RasMol <B>set fontsize 24 PS</B> command, followed by <B>set fontstroke 2</B> <P>
 To display
atom labels on the screen use the RasMol <B>label</B> command, and to change the
colour of displayed labels use the <B>colour labels</B> command. <P>
 </DD>

<DT><B>Set HBonds</B> </DT>
<DD>The
RasMol <B>hbonds</B> parameter determines whether hydrogen bonds are drawn between
the donor and acceptor atoms of the hydrogen bond, <B>set hbonds sidechain</B>
or between the alpha carbon atoms of the protein backbone and between the
phosphorous atoms of the nucleic acid backbone, <B>set hbonds backbone.</B> The
actual display of hydrogen bonds is controlled by the <B>hbonds</B> command. Drawing
hydrogen bonds between protein alpha carbons or nucleic acid phosphorous
atoms is useful when the rest of the molecule is shown in only a schematic
representation such as <B>backbone,</B> <B>ribbons</B> or <B>strands.</B> This parameter is similar
to the RasMol <B>ssbonds</B> parameter. <P>
 </DD>

<DT><B>Set Hetero</B> </DT>
<DD>The RasMol <B>hetero</B> parameter
is used to modify the 'default' behaviour of the RasMol <B>select</B> command, i.e.
the behaviour of <B>select</B> without any parameters. When this value is <B>false,</B>
the default <B>select</B> region does not include any heterogeneous atoms (refer
to the predefined set <B>hetero</B> ). When this value is <B>true,</B> the default <B>select</B>
region may contain hetero atoms. This parameter is similar to the RasMol
<B>hydrogen</B> parameter which determines whether hydrogen atoms should be included
in the default set. If both <B>hetero</B> and <B>hydrogen</B> are <B>true,</B> <B>select</B> without
any parameters is equivalent to <B>select all.</B> <P>
 </DD>

<DT><B>Set HourGlass</B> </DT>
<DD>The RasMol <B>hourglass</B>
parameter allows the user to enable and disable the use of the 'hour glass'
cursor used by RasMol to indicate that the program is currently busy drawing
the next frame. The command <B>set hourglass on</B> enables the indicator, whilst
<B>set hourglass off</B> prevents RasMol from changing the cursor. This is useful
when spinning the molecule, running a sequence of commands from a script
file or using interprocess communication to execute complex sequences of
commands. In these cases a 'flashing' cursor may be distracting. <P>
 </DD>

<DT><B>Set Hydrogen</B>
</DT>
<DD>The RasMol <B>hydrogen</B> parameter is used to modify the "default" behaviour
of the RasMol <B>select</B> command, i.e. the behaviour of <B>select</B> without any parameters.
When this value is <B>false,</B> the default <B>select</B> region does not include any
hydrogen, deuterium or tritium atoms (refer to the predefined set <B>hydrogen</B>
). When this value is <B>true,</B> the default <B>select</B> region may contain hydrogen
atoms. This parameter is similar to the RasMol <B>hetero</B> parameter which determines
whether heterogeneous atoms should be included in the default set. If both
<B>hydrogen</B> and <B>hetero</B> are <B>true,</B> <B>select</B> without any parameters is equivalent
to <B>select all.</B> <P>
 </DD>

<DT><B>Set Kinemage</B> </DT>
<DD>The RasMol <B>set kinemage</B> command controls the
amount of detail stored in a Kinemage output file generated by the RasMol
<B>write kinemage</B> command. The output kinemage files are intended to be displayed
by David Richardson's Mage program. <B>set kinemage false,</B> the default, only
stores the currently displayed representation in the generated output file.
The command <B>set kinemage true,</B> generates a more complex Kinemage that contains
both the wireframe and backbone representations as well as the coordinate
axes, bounding box and crystal unit cell. <P>
 </DD>

<DT><B>Set Menus</B> </DT>
<DD>The RasMol <B>set menus</B>
command enables the canvas window's menu buttons or menu bar. This command
is typically only used by graphical user interfaces or to create as large
an image as possible when using Microsoft Windows. <P>
 </DD>

<DT><B>Set Monitor</B> </DT>
<DD>The RasMol
<B>set monitor</B> command enables <B>monitors.</B> The distance monitor labels may be
turned off with the command <B>set monitor off,</B> and re-enabled with the command
<B>set monitor on.</B> <P>
 <P>
 </DD>

<DT><B>Set Mouse</B> </DT>
<DD>The RasMol <B>set mouse</B> command sets the rotation,
translation, scaling and zooming mouse bindings. The default value is <B>rasmol</B>
which is suitable for two button mice (for three button mice the second
and third buttons are synonymous); X-Y rotation is controlled by the first
button, and X-Y translation by the second. Additional functions are controlled
by holding a modifier key on the keyboard. [Shift] and the first button
performs scaling, [shift] and the second button performs Z-rotation, and
[control] and the first mouse button controls the clipping plane. The <B>insight</B>
and <B>quanta</B> options provide the same mouse bindings as other packages for
experienced users. <P>
 </DD>

<DT><B>Set Picking</B> </DT>
<DD>The RasMol <B>set picking</B> series of commands
affects how a user may interact with a molecule displayed on the screen
in RasMol. <P>
 <B>Enabling/Disabling Atom Identification Picking:</B> Clicking on
an atom with the mouse results in identification and the display of its
residue name, residue number, atom name, atom serial number and chain in
the command window. This behavior may be disabled with the command <B>set picking
none</B> and restored with the command <B>set picking ident.</B> The command <B>set picking
coord</B> adds the atomic coordinates of the atom to the display. <P>
 Disabling
picking, by using <B>set picking off</B> is useful when executing the <B>pause</B> command
in RasMol scripts as it prevents the display of spurious message on the
command line while the script is suspended. <P>
 <B>Measuring Distances, Angles
and Torsions:</B> Interactive measurement of distances, angles and torsions
is achieved using the commands: <B>set picking distance,</B> <B>set picking monitor,</B>
<B>set picking angle</B> and <B>set picking torsion,</B> respectively. In these modes,
clicking on an atom results in it being identified on the rasmol command
line. In addition every atom picked increments a modulo counter such that
in distance mode, every second atom displays the distance (or distance
monitor) between this atom and the previous one. In angle mode, every third
atom displays the angle between the previous three atoms and in torsion
mode every fourth atom displays the torsion between the last four atoms.
By holding down the shift key while picking an atom, this modulo counter
is not incremented and allows, for example, the distances of consecutive
atoms from a fixed atom to be displayed. See the <B>monitor</B> command for how
to control the display of distance monitor lines and labels. <P>
 <B>Labelling
Atoms with the Mouse:</B> The mouse may also be used to toggle the display
of an atom label on a given atom. The RasMol command <B>set picking label</B> removes
a label from a picked atom if it already has one or displays a concise
label at that atom position otherwise. <P>
 <B>Centring Rotation with the Mouse:</B>
A molecule may be centred on a specified atom position using the RasMol
commands <B>set picking centre</B> or <B>set picking center.</B> In this mode, picking
an atom causes all futher rotations to be about that point. <P>
 <B>Picking a Bond
as a Rotation Axis:</B> Any bond may be picked as an axis of rotation for the
portion of the molecule beyond the second atom selected.  This feature should
be used with caution, since, naturally, it changes the conformation of
the molecule.  After executing <B>set picking bond</B> or using the equivalent
"Pick Bond" in the "Settings" menu, a bond to be rotated is picked with
the same sort of mouse clicks as are used for picking atoms for a distance
measurement.  Normally this should be done where a bond exists, but if no
bond exists, it will be added.  The bond cannot be used for rotation if
it is part of a ring of any size.  All bonds selected for rotation are remembered
so that they can be properly reported when writing a script, but only the
most recently selected bond may be actively rotated. <P>
 <B>Enabling Atom/Group/Chain
Selection Picking:</B> Atoms, groups and chains may be selected (as if with
the <B>select</B> command), with the <B>set picking atom,</B> <B>set picking group,</B> <B>set
picking chain</B> commands.  For each of these commands, the shift key may be
used to have a new selection added to the old, and the control key may
be used to have a new selection deleted from the old. When the <B>set picking
atom</B> command is given, the mouse can be used to pick or to drag a box around
the atoms for which selection is desired.  When the <B>set picking group</B> command
is given, picking any an atom will cause selection of all atoms which agree
in residue number with the picked atom, even if in different chains. When
the <B>set picking chain</B> command is given, picking any atom will cause selection
of all atoms which agree in chain identifier with the picked atom. <P>
 </DD>

<DT><B>Set
Radius</B> </DT>
<DD>The RasMol <B>set radius</B> command is used to alter the behaviour of
the RasMol <B>dots</B> command depending upon the value of the <B>solvent</B> parameter.
When <B>solvent</B> is <B>true,</B> the <B>radius</B> parameter controls whether a true van
der Waals' surface is generated by the <B>dots</B> command. If the value of <B>radius</B>
is anything other than zero, that value is used as the radius of each atom
instead of its true vdW value. When the value of <B>solvent</B> is <B>true,</B> this parameter
determines the 'probe sphere' (solvent) radius. The parameter may be given
as an integer in rasmol units or containing a decimal point in Angstroms.
The default value of this parameter is determined by the value of <B>solvent</B>
and changing <B>solvent</B> resets <B>radius</B> to its new default value. <P>
 </DD>

<DT><B>Set ShadePower</B>
</DT>
<DD>The <B>shadepower</B> parameter (adopted from RasTop) determines the shade repartition
(the contrast) used in rendering solid objects. This value between 0 and
100 adjusts shading on an object surface oriented along the direction to
the light source.  Changing the shadepower parameter does not change the
maximum or the minimum values of this shading, as does changing the <B>ambient</B>
parameter.  A value of 100 concentrates the light on the top of spheres,
giving a highly specular, glassy rendering (see the <B>specpower</B> parameter).
A value of 0 distributes the light on the entire object. <P>
 This implementation
of shadepower differs from the one in RasTop only in the choice of range
(0 to 100 versus -20 to 20 in RasTop). <P>
 </DD>

<DT><B>Set Shadow</B> </DT>
<DD>The RasMol <B>set shadow</B>
command enables and disables ray-tracing of the currently rendered image.
Currently only the spacefilling representation is shadowed or can cast
shadows. Enabling shadowing will automatically disable the Z-clipping (slabbing)
plane using the command <B>slab off.</B> Ray-tracing typically takes about several
seconds for a moderately sized protein. It is recommended that shadowing
be normally disabled whilst the molecule is being transformed or manipulated,
and only enabled once an appropiate viewpoint is selected, to provide a
greater impression of depth. <P>
 </DD>

<DT><B>Set SlabMode</B> </DT>
<DD>The RasMol <B>slabmode</B> parameter
controls the rendering method of objects cut by the slabbing (z-clipping)
plane. Valid slabmode parameters are "reject", "half", "hollow", "solid"
and "section". <P>
 </DD>

<DT><B>Set Solvent</B> </DT>
<DD>The RasMol <B>set solvent</B> command is used to control
the behaviour of the RasMol <B>dots</B> command. Depending upon the value of the
<B>solvent</B> parameter, the <B>dots</B> command either generates a van der Waals' or
a solvent accessible surface around the currently selected set of atoms.
Changing this parameter automatically resets the value of the RasMol <B>radius</B>
parameter. The command <B>set solvent false,</B> the default value, indicates that
a van der Waals' surface should be generated and resets the value of <B>radius</B>
to zero. The command <B>set solvent true</B> indicates that a 'Connolly' or 'Richards'
solvent accessible surface should be drawn and sets the <B>radius</B> parameter,
the solvent radius, to 1.2 Angstroms (or 300 RasMol units). <P>
 </DD>

<DT><B>Set Specular</B>
</DT>
<DD>The RasMol <B>set specular</B> command enables and disables the display of specular
highlights on solid objects drawn by RasMol. Specular highlights appear
as white reflections of the light source on the surface of the object. The
current RasMol implementation uses an approximation function to generate
this highlight. <P>
 The specular highlights on the surfaces of solid objects
may be altered by using the specular reflection coefficient, which is altered
using the RasMol <B>set specpower</B> command. <P>
 </DD>

<DT><B>Set SpecPower</B> </DT>
<DD>The <B>specpower</B> parameter
determines the shininess of solid objects rendered by RasMol. This value
between 0 and 100 adjusts the reflection coefficient used in specular highlight
calculations. The specular highlights are enabled and disabled by the RasMol
<B>set specular</B> command. Values around 20 or 30 produce plastic looking surfaces.
High values represent more shiny surfaces such as metals, while lower values
produce more diffuse/dull surfaces. <P>
 </DD>

<DT><B>Set SSBonds</B> </DT>
<DD>The RasMol <B>ssbonds</B> parameter
determines whether disulphide bridges are drawn between the sulphur atoms
in the sidechain (the default) or between the alpha carbon atoms in the
backbone of the cysteines residues. The actual display of disulphide bridges
is controlled by the <B>ssbonds</B> command. Drawing disulphide bridges between
alpha carbons is useful when the rest of the protein is shown in only a
schematic representation such as <B>backbone,</B> <B>ribbons</B> or <B>strands.</B> This parameter
is similar to the RasMol <B>hbonds</B> parameter. <P>
 </DD>

<DT><B>Set Stereo</B> </DT>
<DD>The RasMol <B>set stereo</B>
parameter controls the separation between the left and right images. Turning
stereo on and off doesn't reposition the centre of the molecule. <P>
 Stereo
viewing of a molecule may be turned on (and off) either by selecting <B>Stereo</B>
from the <B>Options</B> menu, or by typing the commands <B>stereo on</B> or <B>stereo off.</B>
<P>
 The separation angle between the two views may be adjusted with the <B>set
stereo [-] &lt;number&gt;</B> command, where positive values result in crossed eye viewing
and negative values in relaxed (wall-eyed) viewing. Currently, stereo viewing
is not supported in <B>vector PostScript</B> output files. <P>
 </DD>

<DT><B>Set Strands</B> </DT>
<DD>The RasMol
<B>strands</B> parameter controls the number of parallel strands that are displayed
in the ribbon representations of proteins. The permissible values for this
parameter are 1, 2, 3, 4, 5 and 9. The default value is 5. The number of
strands is constant for all ribbons being displayed. However, the ribbon
width (the separation between strands) may be controlled on a residue by
residue basis using the RasMol <B>ribbons</B> command. <P>
 </DD>

<DT><B>Set Transparent</B> </DT>
<DD>The RasMol
<B>transparent</B> parameter controls the writing of transparent GIFs by the <B>write
gif &lt;filename&gt;</B> command.  This may be controlled by the <B>set transparent on</B>
and <B>set transparent off</B> commands. <P>
 </DD>

<DT><B>Set UnitCell</B> </DT>
<DD>The RasMol <B>unitcell</B> parameter
controls the display of the crystallographic unit cell on the current display.
The crystal cell is only enabled if the appropriate crystal symmetry information
is contained in the PDB, CIF or mmCIF data file. The RasMol command <B>show
symmetry</B> display details of the crystal's space group and unit cell axes.
The <B>set unitcell</B> command is similar to the commands <B>set axes</B> and <B>set boundbox</B>
that display orthogonal coordinate axes and the bounding box, respectively.
<P>
 </DD>

<DT><B>Set VectPS</B> </DT>
<DD>The RasMol <B>vectps</B> parameter is use to control the way in which
the RasMol <B>write</B> command generates vector PostScript output files. The command
<B>set vectps on</B> enables the use of black outlines around spheres and cylinder
bonds producing "cartoon-like" high resolution output. However, the current
implementation of RasMol incorrectly cartoons spheres that are intersected
by more than one other sphere. Hence "ball and stick" models are rendered
correctly but not large spacefilling spheres models. Cartoon outlines can
be disabled, the default, by the command <B>set vectps off.</B> <P>
 </DD>

<DT><B>Set Write</B> </DT>
<DD>The
RasMol <B>write</B> parameter controls the use of the <B>save</B> and <B>write</B> commands
within scripts, but it may only be executed from the command line.  By default,
this value is <B>false,</B> prohibiting the generation of files in any scripts
executed at start-up (such as those launched from a WWW browser). However,
animators may start up RasMol interactively: type <B>set write on</B> and then
execute a script to generate each frame using the source command. <P>
 </DD>
</DL>

<H2><A NAME="sect7" HREF="#toc7">Atom
Expressions</A></H2>
RasMol atom expressions uniquely identify an arbitrary group
of atoms within a molecule. Atom expressions are composed of either primitive
expressions, predefined sets, comparison operators, <B>within</B> expressions,
or logical (boolean) combinations of the above expression types. <P>
 The logical
operators allow complex queries to be constructed out of simpler ones using
the standard boolean connectives <B>and,</B> <B>or</B> and <B>not.</B> These may be abbreviated
by the symbols "&amp;", "|" and "!", respectively. Parentheses (brackets) may
be used to alter the precedence of the operators. For convenience, a comma
may also be used for boolean disjunction. <P>
 The atom expression is evaluated
for each atom, hence <B>protein and backbone</B> selects protein backbone atoms,
not the protein and [nucleic] acid backbone atoms! <P>
 
<DL>

<DT><B>Primitive Expressions</B>
</DT>
<DD>RasMol primitive expressions are the fundamental building blocks of atom
expressions. There are two types of primitive expression. The first type
is used to identify a given residue number or range of residue numbers.
A single residue is identified by its number (position in the sequence),
and a range is specified by lower and upper bounds separated by a hyphen
character. For example <B>select 5,6,7,8</B> is also <B>select 5-8.</B> Note that this selects
the given residue numbers in all macromolecule chains. <P>
 The second type
of primitive expression specifies a sequence of fields that must match
for a given atom. The first part specifies a residue (or group of residues)
and an optional second part specifies the atoms within those residues. The
first part consists of a residue name, optionally followed by a residue
number and/or chain identifier. <P>
 The second part consists of a period character
followed by an atom name.  An atom name may be up to four alphabetic or
numeric characters. An optional semicolon followed by an alternate conformation
identifier may be appended.  An optional slash followed by a model number
may also be appended. <P>
 An asterisk may be used as a wild card for a whole
field and a question mark as a single character wildcard. <P>
 </DD>

<DT><B>Comparison Operators</B>
</DT>
<DD>Parts of a molecule may also be distinguished using equality, inequality
and ordering operators on their properties. The format of such comparison
expression is a property name, followed by a comparison operator and then
an integer value. <P>
 The atom properties that may be used in RasMol are <B>atomno</B>
for the atom serial number, <B>elemno</B> for the atom's atomic number (element),
<B>resno</B> for the residue number, <B>radius</B> for the spacefill radius in RasMol
units (or zero if not represented as a sphere) and <B>temperature</B> for the
PDB isotropic temperature value. <P>
 The equality operator is denoted either
"=" or "==". The inequality operator as either "&lt;&gt;", "!=" or "/=". The ordering
operators are "&lt;" for less than, "&lt;=" for less than or equal to, "&gt;" for greater
than, and "&gt;" for greater than or equal to. <P>
 </DD>

<DT><B>Within Expressions</B> </DT>
<DD>A RasMol
<B>within</B> expression allows atoms to be selected on their proximity to another
set of atoms. A <B>within</B> expression takes two parameters separated by a comma
and surrounded by parentheses. The first argument is an integer value called
the "cut-off" distance of the within expression and the second argument
is any  valid atom expression. The cut-off distance is expressed in either
integer RasMol units or Angstroms containing a decimal point. An atom is
selected if it is within the cut-off distance of any of the atoms defined
by the second argument. This allows complex expressions to be constructed
containing nested <B>within</B> expressions. <P>
 For example, the command <B>select within(3.2,backbone)</B>
selects any atom within a 3.2 Angstrom radius of any atom in a protein or
nucleic acid backbone. <B>Within</B> expressions are particularly useful for selecting
the atoms around an active site. <P>
 </DD>

<DT><B>Predefined Sets</B> </DT>
<DD>RasMol atom expressions
may contain predefined sets. These sets are single keywords that represent
portions of a molecule of interest. Predefined sets are often abbreviations
of primitive atom expressions. In some cases the use of predefined sets
allows selection of areas of a molecule that could not otherwise be distinguished.
A list of the currently predefined sets is given below. In addition to the
sets listed here, RasMol also treats element names (and their plurals)
as predefined sets containing all atoms of that element type, i.e. the command
<B>select oxygen</B> is equivalent to the command <B>select elemno=8.</B> <P>
 </DD>
</DL>

<H2><A NAME="sect8" HREF="#toc8">Predefined
Sets</A></H2>

<DL>

<DT><B>AT Set</B> </DT>
<DD>This set contains the atoms in the complementary nucleotides
adenosine and thymidine (A and T, respectively). All nucleotides are classified
as either the set <B>at</B> or the set <B>cg</B> This set is equivalent to the RasMol
atom expressions <B>a,t,</B> and <B>nucleic and not cg.</B> <P>
 </DD>

<DT><B>Acidic Set</B> </DT>
<DD>The set of acidic
amino acids. These are the residue types Asp and Glu. All amino acids are
classified as either <B>acidic,</B> <B>basic</B> <B>or</B> <B>neutral.</B> This set is equivalent to
the RasMol atom expressions <B>asp, glu</B> and <B>amino and not (basic or neutral).</B>
<P>
 </DD>

<DT><B>Acyclic Set</B> </DT>
<DD>The set of atoms in amino acids not containing a cycle or
ring. All amino acids are classified as either <B>cyclic</B> or <B>acyclic.</B> This set
is equivalent to the RasMol atom expression <B>amino and not cyclic.</B> <P>
 </DD>

<DT><B>Aliphatic
Set</B> </DT>
<DD>This set contains the aliphatic amino acids. These are the amino acids
Ala, Gly, Ile, Leu and Val. This set is equivalent to the RasMol atom expression
<B>ala, gly, ile, leu, val.</B> <P>
 </DD>

<DT><B>Alpha Set</B> </DT>
<DD>The set of alpha carbons in the protein
molecule. This set is approximately equivalent to the RasMol atom expression
<B>*.CA.</B> This command should not be confused with the predefined set <B>helix</B> which
contains the atoms in the amino acids of the protein's alpha helices. <P>
 </DD>

<DT><B>Amino
Set</B> </DT>
<DD>This set contains all the atoms contained in amino acid residues. This
is useful for distinguishing the protein from the nucleic acid and heterogeneous
atoms in the current molecule database. <P>
 </DD>

<DT><B>Aromatic Set</B> </DT>
<DD>The set of atoms in
amino acids containing aromatic rings. These are the amino acids His, Phe,
Trp and Tyr. Because they contain aromatic rings all members of this set
are member of the predefined set <B>cyclic.</B> This set is equivalent to the RasMol
atom expressions <B>his, phe, trp, tyr</B> and <B>cyclic and not pro.</B> <P>
 </DD>

<DT><B>Backbone Set</B>
</DT>
<DD>This set contains the four atoms of each amino acid that form the polypeptide
N-C-C-O backbone of proteins, and the atoms of the sugar phosphate backbone
of nucleic acids. Use the RasMol predefined sets <B>protein</B> and <B>nucleic</B> to
distinguish between the two forms of backbone. Atoms in nucleic acids and
proteins are either <B>backbone</B> or <B>sidechain.</B> This set is equivalent to the
RasMol expression <B>(protein or nucleic) and not sidechain.</B> <P>
 The predefined
set <B>mainchain</B> is synonymous with the set <B>backbone.</B> <P>
 </DD>

<DT><B>Basic Set</B> </DT>
<DD>The set of
basic amino acids. These are the residue types Arg, His and Lys. All amino
acids are classified as either <B>acidic,</B> <B>basic</B> or <B>neutral.</B> This set is equivalent
to the RasMol atom expressions <B>arg, his, lys</B> and <B>amino and not (acidic
or neutral).</B> <P>
 </DD>

<DT><B>Bonded Set</B> </DT>
<DD>This set contain all the atoms in the current molecule
database that are bonded to at least one other atom. <P>
 </DD>

<DT><B>Buried Set</B> </DT>
<DD>This set
contains the atoms in those amino acids that tend (prefer) to be buried
inside protein, away from contact with solvent molecules. This set refers
to the amino acids preference and not the actual solvent accessibility
for the current protein. All amino acids are classified as either <B>surface</B>
or <B>buried.</B> This set is equivalent to the RasMol atom expression <B>amino and
not surface.</B> <P>
 </DD>

<DT><B>CG Set</B> </DT>
<DD>This set contains the atoms in the complementary nucleotides
cytidine and guanosine (C and G, respectively). All nucleotides are classified
as either the set <B>at</B> or the set <B>cg</B> This set is equivalent to the RasMol
atom expressions <B>c,g</B> and <B>nucleic and not at.</B> <P>
 </DD>

<DT><B>Charged Set</B> </DT>
<DD>This set contains
the charged amino acids. These are the amino acids that are either <B>acidic</B>
or <B>basic.</B> Amino acids are classified as being either <B>charged</B> or <B>neutral.</B>
This set is equivalent to the RasMol atom expressions <B>acidic or basic</B> and
<B>amino and not neutral.</B> <P>
 </DD>

<DT><B>Cyclic Set</B> </DT>
<DD>The set of atoms in amino acids containing
a cycle or rings. All amino acids are classified as either <B>cyclic</B> or <B>acyclic.</B>
This set consists of the amino acids His, Phe, Pro, Trp and Tyr. The members
of the predefined set <B>aromatic</B> are members of this set. The only cyclic
but non-aromatic amino acid is proline. This set is equivalent to the RasMol
atom expressions <B>his, phe, pro, trp, tyr</B> and <B>aromatic or pro</B> and <B>amino
and not acyclic.</B> <P>
 </DD>

<DT><B>Cystine Set</B> </DT>
<DD>This set contains the atoms of cysteine residues
that form part of a disulphide bridge, i.e. half cystines. RasMol automatically
determines disulphide bridges, if neither the predefined set <B>cystine</B> nor
the RasMol <B>ssbonds</B> command have been used since the molecule was loaded.
The set of free cysteines may be determined using the RasMol atom expression
<B>cys and not cystine.</B> <P>
 </DD>

<DT><B>Helix Set</B> </DT>
<DD>This set contains all atoms that form part
of a protein alpha helix as determined by either the PDB file author or
Kabsch and Sander's DSSP algorithm. By default, RasMol uses the secondary
structure determination given in the PDB file if it exists. Otherwise, it
uses the DSSP algorithm as used by the RasMol <B>structure</B> command. <P>
 This predefined
set should not be confused with the predefined set <B>alpha</B> which contains
the alpha carbon atoms of a protein. <P>
 </DD>

<DT><B>Hetero Set</B> </DT>
<DD>This set contains all the
heterogeneous atoms in the molecule. These are the atoms described by HETATM
entries in the PDB file. These typically contain water, cofactors and other
solvents and ligands. All <B>hetero</B> atoms are classified as either <B>ligand</B> or
<B>solvent</B> atoms. These heterogeneous <B>solvent</B> atoms are further classified
as either <B>water</B> or <B>ions.</B> <P>
 </DD>

<DT><B>Hydrogen Set</B> </DT>
<DD>This predefined set contains all
the hydrogen, deuterium and tritium atoms of the current molecule. This
predefined set is equivalent to the RasMol atom expression <B>elemno=1.</B> <P>
 </DD>

<DT><B>Hydrophobic
Set</B> </DT>
<DD>This set contains all the hydrophobic amino acids. These are the amino
acids Ala, Leu, Val, Ile, Pro, Phe, Met and Trp. All amino acids are classified
as either <B>hydrophobic</B> or <B>polar.</B> This set is equivalent to the RasMol atom
expressions <B>ala, leu, val, ile, pro, phe, met, trp</B> and <B>amino and not polar.</B>
<P>
 </DD>

<DT><B>Ions Set</B> </DT>
<DD>This set contains all the heterogeneous phosphate and sulphate
ions in the current molecule data file. A large number of these ions are
sometimes associated with protein and nucleic acid structures determined
by X-ray crystallography. These atoms tend to clutter an image. All <B>hetero</B>
atoms are classified as either <B>ligand</B> or <B>solvent</B> atoms. All <B>solvent</B> atoms
are classified as either <B>water</B> or <B>ions.</B> <P>
 </DD>

<DT><B>Large Set</B> </DT>
<DD>All amino acids are classified
as either <B>small,</B> <B>medium</B> or <B>large.</B> This set is equivalent to the RasMol atom
expression <B>amino and not (small or medium).</B> <P>
 </DD>

<DT><B>Ligand Set</B> </DT>
<DD>This set contains
all the heterogeneous cofactor and ligand moieties that are contained in
the current molecule data file.  This set is defined to be all <B>hetero</B> atoms
that are not <B>solvent</B> atoms. Hence this set is equivalent to the RasMol atom
expression <B>hetero and not solvent.</B> <P>
 </DD>

<DT><B>Medium Set</B> </DT>
<DD>All amino acids are classified
as either <B>small,</B> <B>medium</B> or <B>large.</B> This set is equivalent to the RasMol atom
expression <B>amino and not (large or small).</B> <P>
 </DD>

<DT><B>Neutral Set</B> </DT>
<DD>The set of neutral
amino acids. All amino acids are classified as either <B>acidic,</B> <B>basic</B> or <B>neutral.</B>
This set is equivalent to the RasMol atom expression <B>amino and not (acidic
or basic).</B> <P>
 </DD>

<DT><B>Nucleic Set</B> </DT>
<DD>The set of all atoms in nucleic acids, which consists
of the four nucleotide bases adenosine, cytidine, guanosine and thymidine
(A, C, G and T, respectively). All neucleotides are classified as either
<B>purine</B> or <B>pyrimidine.</B> This set is equivalent to the RasMol atom expressions
<B>a,c,g,t</B> and <B>purine or pyrimidine.</B> The symbols for RNA nucleotides (U, +U,
I, 1MA, 5MC, OMC, 1MG, 2MG, M2G, 7MG, OMG, YG, H2U, 5MU, and PSU) are also
recognized as members of this set. <P>
 </DD>

<DT><B>Polar Set</B> </DT>
<DD>This set contains the polar
amino acids. All amino acids are classified as either <B>hydrophobic</B> or <B>polar.</B>
This set is equivalent to the RasMol atom expression <B>amino and not hydrophobic.</B>
<P>
 </DD>

<DT><B>Protein Set</B> </DT>
<DD>The set of all atoms in proteins. This consists of the RasMol
predefined set <B>amino</B> and common post-translation modifications. <P>
 </DD>

<DT><B>Purine Set</B>
</DT>
<DD>The set of purine nucleotides. These are the bases adenosine and guanosine
(A and G, respectively). All nucleotides are either <B>purines</B> or <B>pyrimidines.</B>
This set is equivalent to the RasMol atom expressions <B>a,g</B> and <B>nucleic and
not pyrimidine.</B> <P>
 </DD>

<DT><B>Pyrimidine Set</B> </DT>
<DD>The set of pyrimidine nucleotides. These
are the bases cytidine and thymidine (C and T, respectively). All nucleotides
are either <B>purines</B> or <B>pyrimidines.</B> This set is equivalent to the RasMol
atom expressions <B>c,t</B> and <B>nucleic and not purine.</B> <P>
 </DD>

<DT><B>Selected Set</B> </DT>
<DD>This set
contains the set of atoms in the currently selected region. The currently
selected region is defined by the preceding <B>select</B> or <B>restrict</B> command
and not the atom expression containing the <B>selected</B> keyword. <P>
 </DD>

<DT><B>Sheet Set</B>
</DT>
<DD>This set contains all atoms that form part of a protein beta sheet as determined
by either the PDB file author or Kabsch and Sander's DSSP algorithm. By default,
RasMol uses the secondary structure determination given in the PDB file
if it exists. Otherwise, it uses the DSSP algorithm as used by the RasMol
<B>structure</B> command. <P>
 </DD>

<DT><B>Sidechain Set</B> </DT>
<DD>This set contains the functional sidechains
of any amino acids and the base of each nucleotide. These are the atoms
not part of the polypeptide N-C-C-O backbone of proteins or the sugar phosphate
backbone of nucleic acids. Use the RasMol predefined sets <B>protein</B> and <B>nucleic</B>
to distinguish between the two forms of sidechain. Atoms in nucleic acids
and proteins are either <B>backbone</B> or <B>sidechain.</B> This set is equivalent to
the RasMol expression <B>(protein or nucleic) and not backbone.</B> <P>
 </DD>

<DT><B>Small Set</B>
</DT>
<DD>All amino acids are classified as either <B>small,</B> <B>medium</B> or <B>large.</B> This set
is equivalent to the RasMol atom expression <B>amino and not (medium or large).</B>
<P>
 </DD>

<DT><B>Solvent Set</B> </DT>
<DD>This set contains the solvent atoms in the molecule coordinate
file. These are the heterogeneous water molecules, phosphate and sulphate
ions. All <B>hetero</B> atoms are classified as either <B>ligand</B> or <B>solvent</B> atoms.
All <B>solvent</B> atoms are classified as either <B>water</B> or <B>ions.</B> This set is equivalent
to the RasMol atom expressions <B>hetero and not ligand</B> and <B>water or ions.</B>
<P>
 </DD>

<DT><B>Surface Set</B> </DT>
<DD>This set contains the atoms in those amino acids that tend
(prefer) to be on the surface of proteins, in contact with solvent molecules.
This set refers to the amino acids preference and not the actual solvent
accessibility for the current protein. All amino acids are classified as
either <B>surface</B> or <B>buried.</B> This set is equivalent to the RasMol atom expression
<B>amino and not buried.</B> <P>
 </DD>

<DT><B>Turn Set</B> </DT>
<DD>This set contains all atoms that form part
of a protein turns as determined by either the PDB file author or Kabsch
and Sander's DSSP algorithm. By default, RasMol uses the secondary structure
determination given in the PDB file if it exists. Otherwise, it uses the
DSSP algorithm as used by the RasMol <B>structure</B> command. <P>
 </DD>

<DT><B>Water Set</B> </DT>
<DD>This
set contains all the heterogeneous water molecules in the current database.
A large number of water molecules are sometimes associated with protein
and nucleic acid structures determined by X-ray crystallography. These atoms
tend to clutter an image. All <B>hetero</B> atoms are classified as either <B>ligand</B>
or <B>solvent</B> atoms. The <B>solvent</B> atoms are further classified as either <B>water</B>
or <B>ions.</B> <P>
 </DD>

<DT><B>Set Summary</B> </DT>
<DD>The table below summarises RasMol's classification
of the common amino acids. <P>
 </DD>
</DL>

<H2><A NAME="sect9" HREF="#toc9">Colour Schemes</A></H2>
The RasMol <B>colour</B> command allows
different objects (such as atoms, bonds and ribbon segments) to be given
a specified colour. Typically this colour is either a RasMol predefined
colour name or an RGB triple. Additionally RasMol also supports <B>alt,</B> <B>amino,</B>
<B>chain,</B> <B>charge,</B> <B>cpk,</B> <B>group,</B> <B>model,</B> <B>shapely,</B> <B>structure,</B> <B>temperature</B> or <B>user</B>
colour schemes for atoms, and <B>hbond type</B> colour scheme for hydrogen bonds
and <B>electrostatic potential</B> colour scheme for dot surfaces. The 24 currently
predefined colour names are Black, Blue, BlueTint, Brown, Cyan, Gold, Grey,
Green, GreenBlue, GreenTint, HotPink, Magenta, Orange, Pink, PinkTint,
Purple, Red, RedOrange, SeaGreen, SkyBlue, Violet, White, Yellow and YellowTint
<P>
 If you frequently wish to use a colour not predefined, you can write a
one-line script. For example, if you make the file <B>grey.col</B> containing the
line, <B>colour [180,180,180] #grey,</B> then the command <B>script grey.col</B> colours
the currently selected atom set grey. <P>
 
<DL>

<DT><B>Alt Colours</B> </DT>
<DD>The RasMol <B>alt</B> (Alternate
Conformer) colour scheme codes the base structure with one colour and applies
a limited number of colours to each alternate conformer.  In a RasMol built
for 8-bit color systems, 4 colours are allowed for alternate conformers.
Otherwise, 8 colours are available. <P>
 </DD>

<DT><B>Amino Colours</B> </DT>
<DD>The RasMol <B>amino</B> colour
scheme colours amino acids according to traditional amino acid properties.
The purpose of colouring is to identify amino acids in an unusual or surprising
environment. The outer parts of a protein that are polar are visible (bright)
colours and non-polar residues darker. Most colours are hallowed by tradition.
This colour scheme is similar to the <B>shapely</B> scheme. <P>
 </DD>

<DT><B>Chain Colours</B> </DT>
<DD>The
RasMol <B>chain</B> colour scheme assigns each macromolecular chain a unique colour.
This colour scheme is particularly useful for distinguishing the parts
of multimeric structure or the individual 'strands' of a DNA chain. <B>Chain</B>
can be selected from the RasMol <B>Colours</B> menu. <P>
 </DD>

<DT><B>Charge Colours</B> </DT>
<DD>The RasMol
<B>charge</B> colour scheme colour codes each atom according to the charge value
stored in the input file (or beta factor field of PDB files). High values
are coloured in blue (positive) and lower values coloured in red (negative).
Rather than use a fixed scale this scheme determines the maximum and minimum
values of the charge/temperature field and interpolates from red to blue
appropriately. Hence, green cannot be assumed to be 'no net charge' charge.
<P>
 The difference between the <B>charge</B> and <B>temperature</B> colour schemes is that
increasing temperature values proceed from blue to red, whereas increasing
charge values go from red to blue. <P>
 If the charge/temperature field stores
reasonable values it is possible to use the RasMol <B>colour dots potential</B>
command to colour code a dot surface (generated by the <B>dots</B> command) by
electrostatic potential. <P>
 </DD>

<DT><B>CPK Colours</B> </DT>
<DD>The RasMol <B>cpk</B> colour scheme is based
upon the colours of the popular plastic spacefilling models which were
developed by Corey, Pauling and later improved by Kultun. This colour scheme
colours 'atom' objects by the atom (element) type. This is the scheme conventionally
used by chemists. The assignment of the most commonly used element types
to colours is given below. <P>
 </DD>

<DT><B>Group Colours</B> </DT>
<DD>The RasMol <B>group</B> colour scheme
colour codes residues by their position in a macromolecular chain. Each
chain is drawn as a smooth spectrum from blue through green, yellow and
orange to red. Hence the N terminus of proteins and 5' terminus of nucleic
acids are coloured red and the C terminus of proteins and 3' terminus of
nucleic acids are drawn in blue. If a chain has a large number of heterogeneous
molecules associated with it, the macromolecule may not be drawn in the
full 'range' of the spectrum. <B>Group</B> can be selected from the RasMol <B>Colours</B>
menu. <P>
 If a chain has a large number of heterogeneous molecules associated
with it, the macromolecule may not be drawn in the full range of the spectrum.
When RasMol performs group coloring it decides the range of colors it uses
from the residue numbering given in the PDB file. Hence the lowest residue
number is displayed in blue and the highest residue number is displayed
as red. Unfortunately, if a PDB file contains a large number of heteroatoms,
such as water molecules, that occupy the high residue numbers, the protein
is displayed in the blue-green end of the spectrum and the waters in the
yellow-red end of the spectrum. This is aggravated by there typically being
many more water molecules than amino acid residues. The solution to this
problem is to use the command <B>set hetero off</B> before applying the group
color scheme. This can also be achieved by toggling <B>Hetero Atoms</B> on the
<B>Options</B> menu before selecting <B>Group</B> on the <B>Colour</B> menu. This command instructs
RasMol to only use non-hetero residues in the group color scaling. <P>
 </DD>

<DT><B>NMR Model
Colours</B> </DT>
<DD>The RasMol <B>model</B> colour scheme codes each NMR model with a distinct
colour.  The NMR model number is taken as a numeric value.  High values are
coloured in blue and lower values coloured in red. Rather than use a fixed
scale this scheme determines the maximum value of the NMR model number
and interpolates from red to blue appropriately. <P>
 </DD>

<DT><B>Shapely Colours</B> </DT>
<DD>The RasMol
<B>shapely</B> colour scheme colour codes residues by amino acid property. This
scheme is based upon Bob Fletterick's "Shapely Models". Each amino acid and
nucleic acid residue is given a unique colour. The <B>shapely</B> colour scheme
is used by David Bacon's Raster3D program. This colour scheme is similar
to the <B>amino</B> colour scheme. <P>
 </DD>

<DT><B>Structure Colours</B> </DT>
<DD>The RasMol <B>structure</B> colour
scheme colours the molecule by protein secondary structure. Alpha helices
are coloured magenta, [240,0,128], beta sheets are coloured yellow, [255,255,0],
turns are coloured pale blue, [96,128,255] and all other residues are coloured
white. The secondary structure is either read from the PDB file (HELIX,
SHEET and TURN records), if available, or determined using Kabsch and Sander's
DSSP algorithm. The RasMol <B>structure</B> command may be used to force DSSP's
structure assignment to be used. <P>
 </DD>

<DT><B>Temperature Colours</B> </DT>
<DD>The RasMol <B>temperature</B>
colour scheme colour codes each atom according to the anisotropic temperature
(beta) value stored in the PDB file. Typically this gives a measure of the
mobility/uncertainty of a given atom's position. High values are coloured
in warmer (red) colours and lower values in colder (blue) colours. This
feature is often used to associate a "scale" value [such as amino acid
variability in viral mutants] with each atom in a PDB file, and colour
the molecule appropriately. <P>
 The difference between the <B>temperature</B> and
<B>charge</B> colour schemes is that increasing temperature values proceed from
blue to red, whereas increasing charge values go from red to blue. <P>
 </DD>

<DT><B>User
Colours</B> </DT>
<DD>The RasMol <B>user</B> colour scheme allows RasMol to use the colour scheme
stored in the PDB file. The colours for each atom are stored in COLO records
placed in the PDB data file. This convention was introduced by David Bacon's
Raster3D program. <P>
 </DD>

<DT><B>HBond Type Colours</B> </DT>
<DD>The RasMol <B>type</B> colour scheme applies
only to hydrogen bonds, hence is used in the command <B>colour hbonds type.</B>
This scheme colour codes each hydrogen bond according to the distance along
a protein chain between hydrogen bond donor and acceptor. This schematic
representation was introduced by Belhadj-Mostefa and Milner-White. This representation
gives a good insight into protein secondary structure (hbonds forming alpha
helices appear red, those forming sheets appear yellow and those forming
turns appear magenta). <P>
 </DD>

<DT><B>Potential Colours</B> </DT>
<DD>The RasMol <B>potential</B> colour scheme
applies only to dot surfaces, hence is used in the command <B>colour dots
potential.</B> This scheme colours each currently displayed dot by the electrostatic
potential at that point in space. This potential is calculated using Coulomb's
law taking the temperature/charge field of the input file to be the charge
assocated with that atom. This is the same interpretation used by the <B>colour
charge</B> command. Like the <B>charge</B> colour scheme low values are blue/white
and high values are red. <P>
 </DD>

<DT><B>Amino Acid Codes</B> </DT>
<DD>The following table lists the
names, single letter and three letter codes of each of the amino acids.
<P>
 </DD>

<DT><B>Booleans</B> </DT>
<DD>A boolean parameter is a truth value. Valid boolean values are
'true' and  used by RasMol to either enable or disable a representation or
option. <P>
 </DD>
</DL>

<H2><A NAME="sect10" HREF="#toc10">File Formats</A></H2>
<P>
 
<DL>

<DT><B>Protein Data Bank Files</B> </DT>
<DD>If you do not have the PDB
documentation, you may find the following summary of the PDB file format
useful. The Protein Data Bank is a computer-based archival database for macromolecular
structures. The database was established in 1971 by Brookhaven National
Laboratory, Upton, New York, as a public domain repository for resolved
crystallographic structures. The Bank uses a uniform format to store atomic
coordinates and partial bond connectivities as derived from crystallographic
studies.  In 1999 the Protein Data Bank moved to the Research Collaboratory
for Structural Biology. <P>
 PDB file entries consist of records of 80 characters
each. Using the punched card analogy, columns 1 to 6 contain a record-type
identifier, the columns 7 to 70 contain data. In older entries, columns
71 to 80 are normally blank, but may contain sequence information added
by library management programs.  In new entries conforming to the 1996 PDB
format, there is other information in those columns. The first four characters
of the record identifier are sufficient to identify the type of record
uniquely, and the syntax of each record is independent of the order of
records within any entry for a particular macromolecule. <P>
 The only record
types that are of major interest to the RasMol program are the ATOM and
HETATM records which describe the position of each atom. ATOM/HETATM records
contain standard atom names and residue abbreviations, along with sequence
identifiers, coordinates in Angstrom units, occupancies and thermal motion
factors. The exact details are given below as a FORTRAN format statement.
 The "fmt" column indicates use of the field in all PDB formats, in the
1992 and earlier formats or in the 1996 and later formats. <P>
 Residues occur
in order starting from the N-terminal residue for proteins and 5'-terminus
for nucleic acids. If the residue sequence is known, certain atom serial
numbers may be omitted to allow for future insertion of any missing atoms.
Within each residue, atoms are ordered in a standard manner, starting with
the backbone (N-C-C-O for proteins) and proceeding in increasing remoteness
from the alpha carbon, along the side chain. <P>
 HETATM records are used to
define post-translational modifications and cofactors associated with the
main molecule. TER records are interpreted as breaks in the main molecule's
backbone. <P>
 If present, RasMol also inspects HEADER, COMPND, HELIX, SHEET,
TURN, CONECT, CRYST1, SCALE, MODEL, ENDMDL, EXPDTA and END records. Information
such as the name, database code, revision date and classification of the
molecule are extracted from HEADER and COMPND records, initial secondary
structure assignments are taken from HELIX, SHEET and TURN records, and
the end of the file may be indicated by an END record. <P>
 </DD>

<DT><B>RasMol Interpretation
of PDB fields</B> </DT>
<DD>Atoms located at 9999.000, 9999.000, 9999.000 are assumed to
be Insight pseudo atoms and are ignored by RasMol. Atom names beginning
' Q' are also assumed to be pseudo atoms or position markers. <P>
 When a data
file contains an NMR structure, multiple conformations may be placed in
a single PDB file delimited by pairs of MODEL and ENDMDL records. RasMol
displays all the NMR models contained in the file. <P>
 Residue names "CSH",
"CYH" and "CSM" are considered pseudonyms for cysteine "CYS". Residue names
"WAT", "H20", "SOL" and "TIP" are considered pseudonyms for water "HOH".
The residue name "D20" is consider heavy water "DOD". The residue name "SUL"
is considered a sulphate ion "SO4". The residue name "CPR" is considered
to be cis-proline and is translated as "PRO". The residue name "TRY" is considered
a pseudonym for tryptophan "TRP". <P>
 RasMol uses the HETATM fields to define
the sets hetero, water, solvent and ligand. Any group with the name "HOH",
"DOD", "SO4" or "PO4" (or aliased to one of these names by the preceding
rules) is considered a solvent and is considered to be defined by a HETATM
field. <P>
 RasMol only respects CONECT connectivity records in PDB files containing
fewer than 256 atoms. This is explained in more detail in the section on
determining molecule connectivity. CONECT records that define a bond more
than once are interpreted as specifying the bond order of that bond, i.e.
a bond specified twice is a double bond and a bond specified three (or
more) times is a triple bond.  This is not a standard PDB feature. <P>
 </DD>

<DT><B>PDB Colour
Scheme Specification</B> </DT>
<DD>RasMol also accepts the supplementary COLO record
type in the PDB files. This record format was introduced by David Bacon's
Raster3D program for specifying the colour scheme to be used when rendering
the molecule. This extension is not currently supported by the PDB. The COLO
record has the same basic record type as the ATOM and HETATM records described
above. <P>
 Colours are assigned to atoms using a matching process. The Mask
field is used in the matching process as follows. First RasMol reads in
and remembers all the ATOM, HETATM and COLO records in input order. When
the user-defined ('User') colour scheme is selected, RasMol goes through each
remembered ATOM/HETATM record in turn, and searches for a COLO record that
matches in all of columns 7 through 30. The first such COLO record to be
found determines the colour and radius of the atom. <P>
 Note that the Red,
Green and Blue components are in the same positions as the X, Y, and Z
components of an ATOM or HETA record, and the van der Waals radius goes
in the place of the Occupancy. The Red, Green and Blue components must all
be in the range 0 to 1. <P>
 In order that one COLO record can provide colour
and radius specifications for more than one atom (e.g. based on residue,
atom type, or any other criterion for which labels can be given somewhere
in columns 7 through 30), a 'don't-care' character, the hash mark "#" (number
or sharp sign) is used. This character, when found in a COLO record, matches
any character in the corresponding column in a ATOM/HETATM record. All other
characters must match identically to count as a match. As an extension to
the specification, any atom that fails to match a COLO record is displayed
in white. <P>
 </DD>

<DT><B>Multiple NMR Models</B> </DT>
<DD>RasMol loads all of the NMR models from a
PDB file no matter which command is used: <B>load pdb &lt;filename&gt;</B> or <B>load nmrpdb
&lt;filename&gt;</B> <P>
 Once multiple NMR conformations have been loaded they may be
manipulated with the atom expression extensions described in <B>Primitive
Expressions.</B> In particular, the command <B>restrict */1</B> will restrict the display
to the first model only. <P>
 </DD>

<DT><B>CIF and mmCIF Format Files</B> </DT>
<DD>CIF is the IUCr standard
for presentation of small molecules and mmCIF is intended as the replacement
for the fixed-field PDB format for presentation of macromolecular structures.
RasMol can accept data sets in either format. <P>
 There are many useful sites
on the World Wide Web where information tools and software related to CIF,
mmCIF and the PDB can be found. The following are good starting points for
exploration: <P>
 The International Union of Crystallography (IUCr) provides
access to software, dictionaries, policy statements and documentation relating
to CIF and mmCIF at: IUCr, Chester, England (www.iucr.org/iucr-top/cif/) with
many mirror sites. <P>
 The Nucleic Acid Database Project provides access to
its entries, software and documentation, with an mmCIF page giving access
to the dictionary and mmCIF software tools at Rutgers University, New Jersey,
USA (http://ndbserver.rutgers.edu/NDB/mmcif) with many mirror sites. <P>
 This
version of RasMol restricts CIF or mmCIF tag values to essentially the
same conventions as are used for the fixed-field PDB format. Thus chain identifiers
and alternate conformation identifiers are limited to a single character,
atom names are limited to 4 characters, etc. RasMol interprets the following
CIF and mmCIF tags: A search is made through multiple data blocks for the
desired tags, so a single dataset may be composed from multiple data blocks,
but multiple data sets may not be stacked in the same file. <P>
 </DD>
</DL>

<H2><A NAME="sect11" HREF="#toc11">Machine-specific
Support</A></H2>
<P>
 In the following sections, support for <B>Monochrome X-Windows,</B> <B>Tcl/Tk
IPC,</B> <B>UNIX sockets based IPC,</B> <B>Compiling RasWin with Borland and MetroWerks</B>
are described. <P>
 
<DL>

<DT><B>Monochrome X-Windows Support</B> </DT>
<DD>RasMol supports the many monochrome
UNIX workstations typically found in academia, such as low-end SUN workstations
and NCD X-terminals. The X11 version of RasMol (when compiled in 8 bit mode)
now detects black and white X-Windows displays and enables dithering automatically.
The use of run-time error diffusion dithering means that all display modes
of RasMol are available when in monochrome mode. For best results, users
should experiment with the set ambient command to ensure the maximum contrast
in resulting images. <P>
 </DD>

<DT><B>Tcl/Tk IPC support</B> </DT>
<DD>Version 4 of Tk graphics library
changed the protocol used to communicate between Tk applications. RasMol
version 2.6 was modified such that it could communicate with both this new
protocol and the previous version 3 protocol supported by RasMol v2.5. Although
Tcl/Tk 3.x applications may only communicate with other 3.x applications
and Tcl/Tk 4.x applications with other 4.x applications, these changes allow
RasMol to communicate between processes with both protocols (potentially
concurrently). <P>
 </DD>

<DT><B>UNIX sockets based IPC</B> </DT>
<DD>The UNIX implementation of RasMol
supports BSD-style socket communication. An identical socket mechanism is
also being developed for VMS, Apple Macintosh and Microsoft Windows systems.
This should allow RasMol to interactively display results of a computation
on a remote host. The current protocol acts as a TCP/IP server on port 21069
that executes command lines until either the command <B>exit</B> or the command
<B>quit</B> is typed. The command exit from the RasMol server, the command <B>quit</B>
both disconnects the current session and terminates RasMol. This functionality
may be tested using the UNIX command <B>telnet &lt;hostname&gt; 21069.</B> <P>
 </DD>

<DT><B>Compiling RasWin
with Borland and MetroWerks</B> </DT>
<DD>A number of changes were made to the source
code in the transition from version 2.5 to 2.6 to allow the Microsoft Windows
version of RasMol to compile using the Borland C/C++ compiler. These fixes
include name changes for the standard library and special code to avoid
a bug in _fmemset. Additional changes were made in the transition from 2.6
to 2.7 to allow compilation with the MetroWerks compilers. <P>
 </DD>
</DL>

<H2><A NAME="sect12" HREF="#toc12">Bibliography</A></H2>
<P>


<DL>

<DT><B>Molecular Graphics</B> </DT>
<DD>[1] Nelson Max, "Computer Representation of Molecular
Surfaces", IEEE Computer Graphics and Applications, pp.21-29, August 1983.
<P>
 [2] Arthur M. Lesk, "Protein Architecture: A Practical Approach", IRL Press
Publishers, 1991. <P>
 </DD>

<DT><B>Molecular Graphics Programs</B> </DT>
<DD>[3] Per J. Kraulis, "MOLSCRIPT:
A Program to Produce both Detailed and Schematic Plots of Protein Structures",
Journal of Applied Crystallography, Vol.24, pp.946-950, 1991. <P>
 [4] David Bacon
and Wayne F. Anderson, "A Fast Algorithm for Rendering Space-Filling Molecule
Pictures", Journal of Molecular Graphics, Vol.6, No.4, pp.219-220, December
1988. <P>
 [5] David C. Richardson and Jane S. Richardson, "The Kinemage: A tool
for Scientific Communication", Protein Science, Vol.1, No.1,pp.3-9, January
1992. <P>
 [6] Mike Carson, "RIBBONS 2.0", Journal of Applied Crystallography,
Vol.24, pp.958-961, 1991. <P>
 [7] Conrad C. Huang, Eric F. Pettersen, Teri E. Klein,
Thomas E. Ferrin and Robert Langridge, "Conic: A Fast Renderer for Space-Filling
Molecules with Shadows", Journal of Molecular Graphics, Vol.9, No.4, pp.230-236,
December 1991. <P>
 </DD>

<DT><B>Molecular Biology Algorithms</B> </DT>
<DD>[8] Wolfgang Kabsch and Christian
Sander, "Dictionary of Protein Secondary Structure: Pattern Recognition
of Hydrogen-Bonded and Geometrical Features", Biopolymers, Vol.22, pp.2577-2637,
1983. <P>
 [9] Michael L. Connolly, "Solvent-Accessible Surfaces of Proteins and
Nucleic Acids", Science, Vol.221, No.4612, pp.709-713, August 1983. <P>
 [10] Khaled
Belhadj-Mostefa, Ron Poet and E. James Milner-White, "Displaying Inter-Main
Chain Hydrogen Bond Patterns in Proteins", Journal of Molecular Graphics,
Vol.9, No.3, pp.194-197, September 1991. <P>
 [11] Mike Carson, "Ribbon Models of
Macromolecules", Journal of Molecular Graphics, Vol.5, No.2, pp.103-106, June
1987. <P>
 [12] Mike Carson and Charles E. Bugg, "Algorithm for Ribbon Models
of Proteins", Journal of Molecular Graphics, Vol.4, No.2, pp.121-122, June
1986. <P>
 [13] H. Iijima, J. B. Dunbar Jr. and G. Marshall, "Calibration of Effective
van der Waals Atomic Contact Radii for Proteins and Peptides", Proteins:
Structure, Functions and Genetics, Vol.2, pp.330-339,1987. <P>
 </DD>

<DT><B>Graphics Algorithms</B>
</DT>
<DD>[14] J. Foley, A. van Dam, S. Feiner and J. Hughes, "Computer Graphics: Principles
and Practice", 2nd Edition, Addison Wesley Publishers, 1990. <P>
 [15] J. Cleary
and G. Wyvill, "Analysis of an Algorithm for Fast Ray Tracing using Uniform
Space Subdivision", The Visual Computer, Vol.4, pp.65-83, 1988. <P>
 [16] Thomas
Porter,"Spherical Shading", Computer Graphics Vol.12, ACM SIGGRAPH, pp.282-285,
1978. <P>
 [17] Jean-Michel Cense, "Exact Visibility Calculation for Space-Filling
Molecular Models", Journal of Molecular Graphics, Vol.9, No.3, pp.191-193,
September 1991. <P>
 [18] Chris Schafmeister, "Fast Algorithm for Generating
CPK Images on Graphics Workstations", Journal of Molecular Graphics, Vol.8,
No.4, pp.201-206, December 1990. <P>
 [19] Bruce A. Johnson, "MSURF: A Rapid and
General Program for the Representation of Molecular Surfaces", Journal
of Molecular Graphics, Vol.5, No.3, pp.167-169, September 1987. <P>
 </DD>

<DT><B>File Formats</B>
</DT>
<DD>[20] Frances C. Bernstein et al., "The Protein Data Bank: A Computer-Based
Archival File for Macromolecular Structures", Journal of Molecular Biology,
Vol.112, pp.535-542, 1977. <P>
 [21] Arthur Dalby, James G. Nourse, W. Douglas Hounshell,
Ann K. I. Gushurst, David L. Grier, Burton A. Leland and John Laufer, "Description
of Several Chemical File Formats Used by Computer Programs Developed at
Molecular Design Limited", Journal of Chemical Information and Computer
Sciences, Vol.32, No.3, pp.244-255, 1992. <P>
 [22] Adobe Systems Inc., "PostScript
Language Reference Manual", Addison-Wesley Publishers, Reading, Mass., 1985.
<P>
 [23] Philip E. Bourne et al., "The Macromolecular Crystallographic Information
File (mmCIF)", Meth. Enzymol. (1997) 277, 571-590. <P>
 [24] Sydney R. Hall, "The
STAR File: a New Format for Electronic Data Transfer and Archiving", Journal
of Chemical Information and Computer Sciences, Vol. 31, 326-333, 1991. <P>
 </DD>
</DL>

<H2><A NAME="sect13" HREF="#toc13">See
Also</A></H2>
The RasMol User Manual! <P>
 
<H2><A NAME="sect14" HREF="#toc14">Author</A></H2>
1992-1998 by Roger Sayle (rasmol@ggr.co.uk)
<P>

<HR><P>
<A NAME="toc"><B>Table of Contents</B></A><P>
<UL>
<LI><A NAME="toc0" HREF="#sect0">Name</A></LI>
<LI><A NAME="toc1" HREF="#sect1">Synopsis</A></LI>
<LI><A NAME="toc2" HREF="#sect2">Formats</A></LI>
<LI><A NAME="toc3" HREF="#sect3">Notices</A></LI>
<LI><A NAME="toc4" HREF="#sect4">Description</A></LI>
<LI><A NAME="toc5" HREF="#sect5">Commands</A></LI>
<LI><A NAME="toc6" HREF="#sect6">Set Parameters</A></LI>
<LI><A NAME="toc7" HREF="#sect7">Atom Expressions</A></LI>
<LI><A NAME="toc8" HREF="#sect8">Predefined Sets</A></LI>
<LI><A NAME="toc9" HREF="#sect9">Colour Schemes</A></LI>
<LI><A NAME="toc10" HREF="#sect10">File Formats</A></LI>
<LI><A NAME="toc11" HREF="#sect11">Machine-specific Support</A></LI>
<LI><A NAME="toc12" HREF="#sect12">Bibliography</A></LI>
<LI><A NAME="toc13" HREF="#sect13">See Also</A></LI>
<LI><A NAME="toc14" HREF="#sect14">Author</A></LI>
</UL>
</BODY></HTML>
