*** cif.c	2004/04/26 20:30:26	1.1
--- cif.c	2004/04/26 20:34:24
***************
*** 1641,1648 ****
          string = c == '\'' || c == '"';
        else
        {
!         string = (cprev != file->text [0] || !isspace(c)) && ( c != '\n' 
!                                                 && c != EOF);
          /* string = c != file->text [0] && c != '\n' && c != EOF; */
          if (!string)
          {
--- 1641,1649 ----
          string = c == '\'' || c == '"';
        else
        {
!         string = (cprev != file->text [0] || file->text_used<2 || 
!                   !isspace(c)) 
!                      && ( c != '\n'  && c != EOF);
          /* string = c != file->text [0] && c != '\n' && c != EOF; */
          if (!string)
          {
