*** infile.c	2004/05/07 19:46:16	1.1
--- infile.c	2004/07/04 20:47:23
***************
*** 1220,1226 ****
  static int FindSybylRefNo( char *ptr )
  {
      register char *src,*dst;
!     auto char name[4];
   
      src = ptr;
      dst = name;
--- 1220,1226 ----
  static int FindSybylRefNo( char *ptr )
  {
      register char *src,*dst;
!     auto char name[5];
   
      src = ptr;
      dst = name;
***************
*** 1252,1257 ****
--- 1252,1258 ----
      {   *dst++ = ' ';
          *dst = ' ';
      }
+     name[4]  = '\0';
      return NewAtomType(name);
  }
   
***************
*** 1336,1344 ****
  {
      register char *ptr;
      register int i;
!     char name[4];
   
      ptr = Record+6;
      if( !isalpha(ptr[1]) )
      {   name[0] = ' ';
          for( i=0; i<3; i++ )
--- 1337,1346 ----
  {
      register char *ptr;
      register int i;
!     char name[5];
   
      ptr = Record+6;
+     name[4] = '\0';
      if( !isalpha(ptr[1]) )
      {   name[0] = ' ';
          for( i=0; i<3; i++ )
