*** molecule.c	2004/05/07 19:46:16	1.1
--- molecule.c	2005/01/21 01:45:07
***************
*** 938,946 ****
  
  int SimpleAtomType( char *type )
  {
!     char name[4];
  
      name[2] = name[3] = ' ';
      if( type[1] && (type[1]!=' ') )
      {   name[0] = ToUpper(type[0]);
          name[1] = ToUpper(type[1]);
--- 955,964 ----
  
  int SimpleAtomType( char *type )
  {
!     char name[5];
  
      name[2] = name[3] = ' ';
+     name[4] = '\0';
      if( type[1] && (type[1]!=' ') )
      {   name[0] = ToUpper(type[0]);
          name[1] = ToUpper(type[1]);
***************
*** 954,962 ****
  
  int ComplexAtomType( char *ptr )
  {
!     static char name[4];
      register int i;
  
      if( isdigit(ptr[1]) )
      {   /* IDATM PDB files! */
          name[0] = ' ';
--- 972,981 ----
  
  int ComplexAtomType( char *ptr )
  {
!     char name[5];
      register int i;
  
+     name[4] ='\0';
      if( isdigit(ptr[1]) )
      {   /* IDATM PDB files! */
          name[0] = ' ';
